// @aws-sdk/client-api-gateway/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareSdkApiGateway = require('@aws-sdk/middleware-sdk-api-gateway');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "apigateway",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class APIGatewayClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(middlewareSdkApiGateway.getAcceptHeaderPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultAPIGatewayHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class APIGatewayServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, APIGatewayServiceException.prototype);
    }
}

class BadRequestException extends APIGatewayServiceException {
    name = "BadRequestException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, BadRequestException.prototype);
    }
}
class ConflictException extends APIGatewayServiceException {
    name = "ConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
class LimitExceededException extends APIGatewayServiceException {
    name = "LimitExceededException";
    $fault = "client";
    retryAfterSeconds;
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
        this.retryAfterSeconds = opts.retryAfterSeconds;
    }
}
class NotFoundException extends APIGatewayServiceException {
    name = "NotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotFoundException.prototype);
    }
}
class TooManyRequestsException extends APIGatewayServiceException {
    name = "TooManyRequestsException";
    $fault = "client";
    retryAfterSeconds;
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.retryAfterSeconds = opts.retryAfterSeconds;
    }
}
class UnauthorizedException extends APIGatewayServiceException {
    name = "UnauthorizedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnauthorizedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnauthorizedException.prototype);
    }
}
class ServiceUnavailableException extends APIGatewayServiceException {
    name = "ServiceUnavailableException";
    $fault = "server";
    retryAfterSeconds;
    constructor(opts) {
        super({
            name: "ServiceUnavailableException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceUnavailableException.prototype);
        this.retryAfterSeconds = opts.retryAfterSeconds;
    }
}

const _A = "Account";
const _AK = "ApiKey";
const _AKI = "ApiKeyIds";
const _AKp = "ApiKeys";
const _ALS = "AccessLogSettings";
const _AS = "ApiStage";
const _Ac = "Accept";
const _Au = "Authorizer";
const _Aut = "Authorizers";
const _BPM = "BasePathMapping";
const _BPMa = "BasePathMappings";
const _BRE = "BadRequestException";
const _CA = "CreateAuthorizer";
const _CAK = "CreateApiKey";
const _CAKR = "CreateApiKeyRequest";
const _CAR = "CreateAuthorizerRequest";
const _CBPM = "CreateBasePathMapping";
const _CBPMR = "CreateBasePathMappingRequest";
const _CC = "ClientCertificate";
const _CCl = "ClientCertificates";
const _CD = "Content-Disposition";
const _CDN = "CreateDomainName";
const _CDNAA = "CreateDomainNameAccessAssociation";
const _CDNAAR = "CreateDomainNameAccessAssociationRequest";
const _CDNR = "CreateDomainNameRequest";
const _CDP = "CreateDocumentationPart";
const _CDPR = "CreateDocumentationPartRequest";
const _CDR = "CreateDeploymentRequest";
const _CDV = "CreateDocumentationVersion";
const _CDVR = "CreateDocumentationVersionRequest";
const _CDr = "CreateDeployment";
const _CE = "ConflictException";
const _CM = "CreateModel";
const _CMR = "CreateModelRequest";
const _CR = "CreateResource";
const _CRA = "CreateRestApi";
const _CRAR = "CreateRestApiRequest";
const _CRR = "CreateResourceRequest";
const _CRV = "CreateRequestValidator";
const _CRVR = "CreateRequestValidatorRequest";
const _CS = "CanarySettings";
const _CSR = "CreateStageRequest";
const _CSr = "CreateStage";
const _CT = "Content-Type";
const _CUP = "CreateUsagePlan";
const _CUPK = "CreateUsagePlanKey";
const _CUPKR = "CreateUsagePlanKeyRequest";
const _CUPR = "CreateUsagePlanRequest";
const _CVL = "CreateVpcLink";
const _CVLR = "CreateVpcLinkRequest";
const _D = "Deployment";
const _DA = "DeleteAuthorizer";
const _DAK = "DeleteApiKey";
const _DAKR = "DeleteApiKeyRequest";
const _DAR = "DeleteAuthorizerRequest";
const _DBPM = "DeleteBasePathMapping";
const _DBPMR = "DeleteBasePathMappingRequest";
const _DCC = "DeleteClientCertificate";
const _DCCR = "DeleteClientCertificateRequest";
const _DCS = "DeploymentCanarySettings";
const _DD = "DeleteDeployment";
const _DDN = "DeleteDomainName";
const _DDNAA = "DeleteDomainNameAccessAssociation";
const _DDNAAR = "DeleteDomainNameAccessAssociationRequest";
const _DDNR = "DeleteDomainNameRequest";
const _DDP = "DeleteDocumentationPart";
const _DDPR = "DeleteDocumentationPartRequest";
const _DDR = "DeleteDeploymentRequest";
const _DDV = "DeleteDocumentationVersion";
const _DDVR = "DeleteDocumentationVersionRequest";
const _DGR = "DeleteGatewayResponse";
const _DGRR = "DeleteGatewayResponseRequest";
const _DI = "DeleteIntegration";
const _DIR = "DeleteIntegrationRequest";
const _DIRR = "DeleteIntegrationResponseRequest";
const _DIRe = "DeleteIntegrationResponse";
const _DM = "DeleteMethod";
const _DMR = "DeleteMethodRequest";
const _DMRR = "DeleteMethodResponseRequest";
const _DMRe = "DeleteModelRequest";
const _DMRel = "DeleteMethodResponse";
const _DMe = "DeleteModel";
const _DN = "DomainName";
const _DNAA = "DomainNameAccessAssociation";
const _DNAAo = "DomainNameAccessAssociations";
const _DNo = "DomainNames";
const _DP = "DocumentationPart";
const _DPI = "DocumentationPartIds";
const _DPL = "DocumentationPartLocation";
const _DPo = "DocumentationParts";
const _DR = "DeleteResource";
const _DRA = "DeleteRestApi";
const _DRAR = "DeleteRestApiRequest";
const _DRR = "DeleteResourceRequest";
const _DRV = "DeleteRequestValidator";
const _DRVR = "DeleteRequestValidatorRequest";
const _DS = "DeleteStage";
const _DSR = "DeleteStageRequest";
const _DUP = "DeleteUsagePlan";
const _DUPK = "DeleteUsagePlanKey";
const _DUPKR = "DeleteUsagePlanKeyRequest";
const _DUPR = "DeleteUsagePlanRequest";
const _DV = "DocumentationVersion";
const _DVL = "DeleteVpcLink";
const _DVLR = "DeleteVpcLinkRequest";
const _DVo = "DocumentationVersions";
const _De = "Deployments";
const _EC = "EndpointConfiguration";
const _ER = "ExportResponse";
const _FSAC = "FlushStageAuthorizersCache";
const _FSACR = "FlushStageAuthorizersCacheRequest";
const _FSC = "FlushStageCache";
const _FSCR = "FlushStageCacheRequest";
const _GA = "GetAccount";
const _GAK = "GetApiKey";
const _GAKR = "GetApiKeyRequest";
const _GAKRe = "GetApiKeysRequest";
const _GAKe = "GetApiKeys";
const _GAR = "GetAccountRequest";
const _GARe = "GetAuthorizerRequest";
const _GARet = "GetAuthorizersRequest";
const _GAe = "GetAuthorizer";
const _GAet = "GetAuthorizers";
const _GBPM = "GetBasePathMapping";
const _GBPMR = "GetBasePathMappingRequest";
const _GBPMRe = "GetBasePathMappingsRequest";
const _GBPMe = "GetBasePathMappings";
const _GCC = "GenerateClientCertificate";
const _GCCR = "GenerateClientCertificateRequest";
const _GCCRe = "GetClientCertificateRequest";
const _GCCRet = "GetClientCertificatesRequest";
const _GCCe = "GetClientCertificate";
const _GCCet = "GetClientCertificates";
const _GD = "GetDeployment";
const _GDN = "GetDomainName";
const _GDNAA = "GetDomainNameAccessAssociations";
const _GDNAAR = "GetDomainNameAccessAssociationsRequest";
const _GDNR = "GetDomainNameRequest";
const _GDNRe = "GetDomainNamesRequest";
const _GDNe = "GetDomainNames";
const _GDP = "GetDocumentationPart";
const _GDPR = "GetDocumentationPartRequest";
const _GDPRe = "GetDocumentationPartsRequest";
const _GDPe = "GetDocumentationParts";
const _GDR = "GetDeploymentRequest";
const _GDRe = "GetDeploymentsRequest";
const _GDV = "GetDocumentationVersion";
const _GDVR = "GetDocumentationVersionRequest";
const _GDVRe = "GetDocumentationVersionsRequest";
const _GDVe = "GetDocumentationVersions";
const _GDe = "GetDeployments";
const _GE = "GetExport";
const _GER = "GetExportRequest";
const _GGR = "GetGatewayResponse";
const _GGRR = "GetGatewayResponseRequest";
const _GGRRe = "GetGatewayResponsesRequest";
const _GGRe = "GetGatewayResponses";
const _GI = "GetIntegration";
const _GIR = "GetIntegrationRequest";
const _GIRR = "GetIntegrationResponseRequest";
const _GIRe = "GetIntegrationResponse";
const _GM = "GetMethod";
const _GMR = "GetMethodRequest";
const _GMRR = "GetMethodResponseRequest";
const _GMRe = "GetModelRequest";
const _GMRet = "GetModelsRequest";
const _GMRete = "GetMethodResponse";
const _GMT = "GetModelTemplate";
const _GMTR = "GetModelTemplateRequest";
const _GMe = "GetModel";
const _GMet = "GetModels";
const _GR = "GatewayResponse";
const _GRA = "GetRestApi";
const _GRAR = "GetRestApiRequest";
const _GRARe = "GetRestApisRequest";
const _GRAe = "GetRestApis";
const _GRR = "GetResourceRequest";
const _GRRe = "GetResourcesRequest";
const _GRV = "GetRequestValidator";
const _GRVR = "GetRequestValidatorRequest";
const _GRVRe = "GetRequestValidatorsRequest";
const _GRVe = "GetRequestValidators";
const _GRa = "GatewayResponses";
const _GRe = "GetResource";
const _GRet = "GetResources";
const _GS = "GetSdk";
const _GSR = "GetSdkRequest";
const _GSRe = "GetStageRequest";
const _GSRet = "GetStagesRequest";
const _GST = "GetSdkType";
const _GSTR = "GetSdkTypeRequest";
const _GSTRe = "GetSdkTypesRequest";
const _GSTe = "GetSdkTypes";
const _GSe = "GetStage";
const _GSet = "GetStages";
const _GT = "GetTags";
const _GTR = "GetTagsRequest";
const _GU = "GetUsage";
const _GUP = "GetUsagePlan";
const _GUPK = "GetUsagePlanKey";
const _GUPKR = "GetUsagePlanKeyRequest";
const _GUPKRe = "GetUsagePlanKeysRequest";
const _GUPKe = "GetUsagePlanKeys";
const _GUPR = "GetUsagePlanRequest";
const _GUPRe = "GetUsagePlansRequest";
const _GUPe = "GetUsagePlans";
const _GUR = "GetUsageRequest";
const _GVL = "GetVpcLink";
const _GVLR = "GetVpcLinkRequest";
const _GVLRe = "GetVpcLinksRequest";
const _GVLe = "GetVpcLinks";
const _I = "Integration";
const _IAK = "ImportApiKeys";
const _IAKR = "ImportApiKeysRequest";
const _IDP = "ImportDocumentationParts";
const _IDPR = "ImportDocumentationPartsRequest";
const _IR = "IntegrationResponse";
const _IRA = "ImportRestApi";
const _IRAR = "ImportRestApiRequest";
const _LEE = "LimitExceededException";
const _LOA = "ListOfAuthorizer";
const _LOAK = "ListOfApiKey";
const _LOAS = "ListOfApiStage";
const _LOBPM = "ListOfBasePathMapping";
const _LOCC = "ListOfClientCertificate";
const _LOD = "ListOfDeployment";
const _LODN = "ListOfDomainName";
const _LODNAA = "ListOfDomainNameAccessAssociation";
const _LODP = "ListOfDocumentationPart";
const _LODV = "ListOfDocumentationVersion";
const _LOGR = "ListOfGatewayResponse";
const _LOM = "ListOfModel";
const _LOPO = "ListOfPatchOperation";
const _LOR = "ListOfResource";
const _LORA = "ListOfRestApi";
const _LORV = "ListOfRequestValidator";
const _LOS = "ListOfStage";
const _LOSCP = "ListOfSdkConfigurationProperty";
const _LOSK = "ListOfStageKeys";
const _LOST = "ListOfSdkType";
const _LOU = "ListOfUsage";
const _LOUP = "ListOfUsagePlan";
const _LOUPK = "ListOfUsagePlanKey";
const _LOVL = "ListOfVpcLink";
const _M = "Method";
const _MOASTS = "MapOfApiStageThrottleSettings";
const _MOIR = "MapOfIntegrationResponse";
const _MOKU = "MapOfKeyUsages";
const _MOM = "MapOfMethod";
const _MOMR = "MapOfMethodResponse";
const _MOMS = "MapOfMethodSettings";
const _MOMSa = "MapOfMethodSnapshot";
const _MOSTL = "MapOfStringToList";
const _MR = "MethodResponse";
const _MS = "MethodSetting";
const _MSe = "MethodSnapshot";
const _MTA = "MutualTlsAuthentication";
const _MTAI = "MutualTlsAuthenticationInput";
const _Mo = "Model";
const _Mod = "Models";
const _NFE = "NotFoundException";
const _PGR = "PutGatewayResponse";
const _PGRR = "PutGatewayResponseRequest";
const _PI = "PutIntegration";
const _PIR = "PutIntegrationRequest";
const _PIRR = "PutIntegrationResponseRequest";
const _PIRu = "PutIntegrationResponse";
const _PM = "PutMethod";
const _PMR = "PutMethodRequest";
const _PMRR = "PutMethodResponseRequest";
const _PMRu = "PutMethodResponse";
const _PO = "PatchOperation";
const _PRA = "PutRestApi";
const _PRAR = "PutRestApiRequest";
const _PTMOMS = "PathToMapOfMethodSnapshot";
const _QS = "QuotaSettings";
const _R = "Resource";
const _RA = "Retry-After";
const _RAe = "RestApi";
const _RAes = "RestApis";
const _RDNAA = "RejectDomainNameAccessAssociation";
const _RDNAAR = "RejectDomainNameAccessAssociationRequest";
const _RV = "RequestValidator";
const _RVe = "RequestValidators";
const _Re = "Resources";
const _S = "Stage";
const _SCP = "SdkConfigurationProperty";
const _SK = "StageKey";
const _SR = "SdkResponse";
const _ST = "SdkType";
const _STd = "SdkTypes";
const _SUE = "ServiceUnavailableException";
const _St = "Stages";
const _T = "Tags";
const _TC = "TlsConfig";
const _TIA = "TestInvokeAuthorizer";
const _TIAR = "TestInvokeAuthorizerRequest";
const _TIARe = "TestInvokeAuthorizerResponse";
const _TIM = "TestInvokeMethod";
const _TIMR = "TestInvokeMethodRequest";
const _TIMRe = "TestInvokeMethodResponse";
const _TMRE = "TooManyRequestsException";
const _TR = "TagResource";
const _TRR = "TagResourceRequest";
const _TS = "ThrottleSettings";
const _Te = "Template";
const _U = "Usage";
const _UA = "UpdateAccount";
const _UAK = "UpdateApiKey";
const _UAKR = "UpdateApiKeyRequest";
const _UAR = "UpdateAccountRequest";
const _UARp = "UpdateAuthorizerRequest";
const _UAp = "UpdateAuthorizer";
const _UBPM = "UpdateBasePathMapping";
const _UBPMR = "UpdateBasePathMappingRequest";
const _UCC = "UpdateClientCertificate";
const _UCCR = "UpdateClientCertificateRequest";
const _UD = "UpdateDeployment";
const _UDN = "UpdateDomainName";
const _UDNR = "UpdateDomainNameRequest";
const _UDP = "UpdateDocumentationPart";
const _UDPR = "UpdateDocumentationPartRequest";
const _UDR = "UpdateDeploymentRequest";
const _UDV = "UpdateDocumentationVersion";
const _UDVR = "UpdateDocumentationVersionRequest";
const _UE = "UnauthorizedException";
const _UGR = "UpdateGatewayResponse";
const _UGRR = "UpdateGatewayResponseRequest";
const _UI = "UpdateIntegration";
const _UIR = "UpdateIntegrationRequest";
const _UIRR = "UpdateIntegrationResponseRequest";
const _UIRp = "UpdateIntegrationResponse";
const _UM = "UpdateMethod";
const _UMR = "UpdateMethodRequest";
const _UMRR = "UpdateMethodResponseRequest";
const _UMRp = "UpdateModelRequest";
const _UMRpd = "UpdateMethodResponse";
const _UMp = "UpdateModel";
const _UP = "UsagePlan";
const _UPK = "UsagePlanKey";
const _UPKs = "UsagePlanKeys";
const _UPs = "UsagePlans";
const _UR = "UntagResource";
const _URA = "UpdateRestApi";
const _URAR = "UpdateRestApiRequest";
const _URR = "UntagResourceRequest";
const _URRp = "UpdateResourceRequest";
const _URV = "UpdateRequestValidator";
const _URVR = "UpdateRequestValidatorRequest";
const _URp = "UpdateResource";
const _US = "UpdateStage";
const _USR = "UpdateStageRequest";
const _UU = "UpdateUsage";
const _UUP = "UpdateUsagePlan";
const _UUPR = "UpdateUsagePlanRequest";
const _UUR = "UpdateUsageRequest";
const _UVL = "UpdateVpcLink";
const _UVLR = "UpdateVpcLinkRequest";
const _VL = "VpcLink";
const _VLp = "VpcLinks";
const _a = "accepts";
const _aAS = "accessAssociationSource";
const _aAST = "accessAssociationSourceType";
const _aC = "authorizerCredentials";
const _aCd = "additionalContext";
const _aI = "apiId";
const _aIu = "authorizerId";
const _aK = "apiKey";
const _aKR = "apiKeyRequired";
const _aKS = "apiKeySource";
const _aKV = "apiKeyVersion";
const _aLS = "accessLogSettings";
const _aRTIS = "authorizerResultTtlInSeconds";
const _aS = "apiStages";
const _aSM = "apiStatusMessage";
const _aSp = "apiSummary";
const _aSpi = "apiStatus";
const _aSu = "authorizationScopes";
const _aT = "authType";
const _aTu = "authorizationType";
const _aU = "authorizerUri";
const _au = "authorization";
const _b = "body";
const _bL = "burstLimit";
const _bMT = "binaryMediaTypes";
const _bP = "basePath";
const _c = "client";
const _cA = "certificateArn";
const _cB = "certificateBody";
const _cC = "certificateChain";
const _cCE = "cacheClusterEnabled";
const _cCI = "clientCertificateId";
const _cCS = "cacheClusterSize";
const _cCSa = "cacheClusterStatus";
const _cD = "createdDate";
const _cDE = "cacheDataEncrypted";
const _cDo = "contentDisposition";
const _cE = "cachingEnabled";
const _cF = "cloneFrom";
const _cH = "contentHandling";
const _cI = "customerId";
const _cIo = "connectionId";
const _cKP = "cacheKeyParameters";
const _cN = "certificateName";
const _cNa = "cacheNamespace";
const _cP = "configurationProperties";
const _cPK = "certificatePrivateKey";
const _cRA = "cloudwatchRoleArn";
const _cS = "canarySettings";
const _cSl = "clientStatus";
const _cT = "contentType";
const _cTIS = "cacheTtlInSeconds";
const _cTo = "connectionType";
const _cUD = "certificateUploadDate";
const _cl = "claims";
const _cr = "credentials";
const _d = "description";
const _dA = "destinationArn";
const _dDN = "distributionDomainName";
const _dEAE = "disableExecuteApiEndpoint";
const _dHZI = "distributionHostedZoneId";
const _dI = "deploymentId";
const _dN = "domainName";
const _dNA = "domainNameArn";
const _dNAAA = "domainNameAccessAssociationArn";
const _dNI = "domainNameId";
const _dNS = "domainNameStatus";
const _dNSM = "domainNameStatusMessage";
const _dPI = "documentationPartId";
const _dR = "defaultResponse";
const _dTE = "dataTraceEnabled";
const _dV = "documentationVersion";
const _dVe = "defaultValue";
const _e = "enabled";
const _eAM = "endpointAccessMode";
const _eC = "endpointConfiguration";
const _eD = "expirationDate";
const _eDn = "endDate";
const _eT = "exportType";
const _em = "embed";
const _er = "error";
const _f = "format";
const _fN = "friendlyName";
const _fOW = "failOnWarnings";
const _fa = "failonwarnings";
const _fe = "features";
const _fl = "flatten";
const _fr = "from";
const _gDI = "generateDistinctId";
const _h = "headers";
const _hE = "httpError";
const _hH = "httpHeader";
const _hL = "httpLabel";
const _hM = "httpMethod";
const _hQ = "httpQuery";
const _ht = "http";
const _i = "id";
const _iAT = "ipAddressType";
const _iHM = "integrationHttpMethod";
const _iR = "integrationResponses";
const _iS = "identitySource";
const _iSV = "insecureSkipVerification";
const _iT = "integrationTarget";
const _iV = "includeValue";
const _iVE = "identityValidationExpression";
const _iVn = "includeValues";
const _id = "ids";
const _it = "items";
const _ite = "item";
const _jN = "jsonName";
const _kI = "keyId";
const _kT = "keyType";
const _l = "location";
const _lL = "loggingLevel";
const _lS = "locationStatus";
const _lUD = "lastUpdatedDate";
const _la = "latency";
const _li = "limit";
const _lo = "log";
const _m = "message";
const _mCS = "minimumCompressionSize";
const _mE = "metricsEnabled";
const _mI = "methodIntegration";
const _mN = "modelName";
const _mP = "managementPolicy";
const _mR = "methodResponses";
const _mS = "methodSettings";
const _mTA = "mutualTlsAuthentication";
const _mVH = "multiValueHeaders";
const _me = "method";
const _mo = "mode";
const _n = "name";
const _nQ = "nameQuery";
const _o = "op";
const _oN = "operationName";
const _oVCA = "ownershipVerificationCertificateArn";
const _of = "offset";
const _p = "position";
const _pARN = "providerARNs";
const _pB = "passthroughBehavior";
const _pC = "productCode";
const _pEC = "pemEncodedCertificate";
const _pI = "parentId";
const _pIr = "principalId";
const _pO = "patchOperations";
const _pP = "pathPart";
const _pT = "percentTraffic";
const _pWQS = "pathWithQueryString";
const _pa = "path";
const _par = "parameters";
const _pe = "period";
const _po = "policy";
const _pr = "properties";
const _q = "quota";
const _r = "required";
const _rA = "resourceArn";
const _rAFCC = "requireAuthorizationForCacheControl";
const _rAI = "restApiId";
const _rAS = "retryAfterSeconds";
const _rCA = "regionalCertificateArn";
const _rCN = "regionalCertificateName";
const _rDN = "regionalDomainName";
const _rHM = "requestHttpMethod";
const _rHZI = "regionalHostedZoneId";
const _rI = "resourceId";
const _rL = "rateLimit";
const _rM = "routingMode";
const _rMe = "requestModels";
const _rMes = "responseModels";
const _rMeso = "resourceMethods";
const _rO = "resourceOwner";
const _rP = "responseParameters";
const _rPe = "requestParameters";
const _rRI = "rootResourceId";
const _rT = "responseType";
const _rTM = "responseTransferMode";
const _rTe = "responseTemplates";
const _rTeq = "requestTemplates";
const _rVI = "requestValidatorId";
const _s = "stage";
const _sC = "statusCode";
const _sD = "stageDescription";
const _sDt = "startDate";
const _sK = "stageKeys";
const _sM = "statusMessage";
const _sN = "stageName";
const _sP = "securityPolicy";
const _sPe = "selectionPattern";
const _sT = "sdkType";
const _sV = "stageVariables";
const _sVO = "stageVariableOverrides";
const _sc = "schema";
const _se = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.apigateway";
const _st = "status";
const _t = "tags";
const _tA = "targetArns";
const _tBL = "throttlingBurstLimit";
const _tC = "tlsConfig";
const _tE = "tracingEnabled";
const _tIM = "timeoutInMillis";
const _tK = "tagKeys";
const _tRL = "throttlingRateLimit";
const _tS = "throttleSettings";
const _tU = "truststoreUri";
const _tV = "truststoreVersion";
const _tW = "truststoreWarnings";
const _th = "throttle";
const _ty = "type";
const _typ = "types";
const _u = "uri";
const _uCCHS = "unauthorizedCacheControlHeaderStrategy";
const _uPI = "usagePlanId";
const _uSC = "useStageCache";
const _v = "value";
const _vEI = "vpcEndpointIds";
const _vLI = "vpcLinkId";
const _vRB = "validateRequestBody";
const _vRP = "validateRequestParameters";
const _va = "variables";
const _val = "values";
const _ve = "version";
const _w = "warnings";
const _wAA = "webAclArn";
const n0 = "com.amazonaws.apigateway";
var AccessLogSettings$ = [3, n0, _ALS,
    0,
    [_f, _dA],
    [0, 0]
];
var Account$ = [3, n0, _A,
    0,
    [_cRA, _tS, _fe, _aKV],
    [0, () => ThrottleSettings$, 64 | 0, 0]
];
var ApiKey$ = [3, n0, _AK,
    0,
    [_i, _v, _n, _cI, _d, _e, _cD, _lUD, _sK, _t],
    [0, 0, 0, 0, 0, 2, 4, 4, 64 | 0, 128 | 0]
];
var ApiKeyIds$ = [3, n0, _AKI,
    0,
    [_id, _w],
    [64 | 0, 64 | 0]
];
var ApiKeys$ = [3, n0, _AKp,
    0,
    [_w, _it, _p],
    [64 | 0, [() => ListOfApiKey, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var ApiStage$ = [3, n0, _AS,
    0,
    [_aI, _s, _th],
    [0, 0, () => MapOfApiStageThrottleSettings]
];
var Authorizer$ = [3, n0, _Au,
    0,
    [_i, _n, _ty, _pARN, _aT, _aU, _aC, _iS, _iVE, _aRTIS],
    [0, 0, 0, 64 | 0, 0, 0, 0, 0, 0, 1]
];
var Authorizers$ = [3, n0, _Aut,
    0,
    [_it, _p],
    [[() => ListOfAuthorizer, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var BadRequestException$ = [-3, n0, _BRE,
    { [_er]: _c, [_hE]: 400 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(BadRequestException$, BadRequestException);
var BasePathMapping$ = [3, n0, _BPM,
    0,
    [_bP, _rAI, _s],
    [0, 0, 0]
];
var BasePathMappings$ = [3, n0, _BPMa,
    0,
    [_it, _p],
    [[() => ListOfBasePathMapping, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var CanarySettings$ = [3, n0, _CS,
    0,
    [_pT, _dI, _sVO, _uSC],
    [1, 0, 128 | 0, 2]
];
var ClientCertificate$ = [3, n0, _CC,
    0,
    [_cCI, _d, _pEC, _cD, _eD, _t],
    [0, 0, 0, 4, 4, 128 | 0]
];
var ClientCertificates$ = [3, n0, _CCl,
    0,
    [_it, _p],
    [[() => ListOfClientCertificate, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var ConflictException$ = [-3, n0, _CE,
    { [_er]: _c, [_hE]: 409 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(ConflictException$, ConflictException);
var CreateApiKeyRequest$ = [3, n0, _CAKR,
    0,
    [_n, _d, _e, _gDI, _v, _sK, _cI, _t],
    [0, 0, 2, 2, 0, () => ListOfStageKeys, 0, 128 | 0]
];
var CreateAuthorizerRequest$ = [3, n0, _CAR,
    0,
    [_rAI, _n, _ty, _pARN, _aT, _aU, _aC, _iS, _iVE, _aRTIS],
    [[0, 1], 0, 0, 64 | 0, 0, 0, 0, 0, 0, 1], 3
];
var CreateBasePathMappingRequest$ = [3, n0, _CBPMR,
    0,
    [_dN, _rAI, _dNI, _bP, _s],
    [[0, 1], 0, [0, { [_hQ]: _dNI }], 0, 0], 2
];
var CreateDeploymentRequest$ = [3, n0, _CDR,
    0,
    [_rAI, _sN, _sD, _d, _cCE, _cCS, _va, _cS, _tE],
    [[0, 1], 0, 0, 0, 2, 0, 128 | 0, () => DeploymentCanarySettings$, 2], 1
];
var CreateDocumentationPartRequest$ = [3, n0, _CDPR,
    0,
    [_rAI, _l, _pr],
    [[0, 1], () => DocumentationPartLocation$, 0], 3
];
var CreateDocumentationVersionRequest$ = [3, n0, _CDVR,
    0,
    [_rAI, _dV, _sN, _d],
    [[0, 1], 0, 0, 0], 2
];
var CreateDomainNameAccessAssociationRequest$ = [3, n0, _CDNAAR,
    0,
    [_dNA, _aAST, _aAS, _t],
    [0, 0, 0, 128 | 0], 3
];
var CreateDomainNameRequest$ = [3, n0, _CDNR,
    0,
    [_dN, _cN, _cB, _cPK, _cC, _cA, _rCN, _rCA, _eC, _t, _sP, _eAM, _mTA, _oVCA, _po, _rM],
    [0, 0, 0, 0, 0, 0, 0, 0, () => EndpointConfiguration$, 128 | 0, 0, 0, () => MutualTlsAuthenticationInput$, 0, 0, 0], 1
];
var CreateModelRequest$ = [3, n0, _CMR,
    0,
    [_rAI, _n, _cT, _d, _sc],
    [[0, 1], 0, 0, 0, 0], 3
];
var CreateRequestValidatorRequest$ = [3, n0, _CRVR,
    0,
    [_rAI, _n, _vRB, _vRP],
    [[0, 1], 0, 2, 2], 1
];
var CreateResourceRequest$ = [3, n0, _CRR,
    0,
    [_rAI, _pI, _pP],
    [[0, 1], [0, 1], 0], 3
];
var CreateRestApiRequest$ = [3, n0, _CRAR,
    0,
    [_n, _d, _ve, _cF, _bMT, _mCS, _aKS, _eC, _po, _t, _dEAE, _sP, _eAM],
    [0, 0, 0, 0, 64 | 0, 1, 0, () => EndpointConfiguration$, 0, 128 | 0, 2, 0, 0], 1
];
var CreateStageRequest$ = [3, n0, _CSR,
    0,
    [_rAI, _sN, _dI, _d, _cCE, _cCS, _va, _dV, _cS, _tE, _t],
    [[0, 1], 0, 0, 0, 2, 0, 128 | 0, 0, () => CanarySettings$, 2, 128 | 0], 3
];
var CreateUsagePlanKeyRequest$ = [3, n0, _CUPKR,
    0,
    [_uPI, _kI, _kT],
    [[0, 1], 0, 0], 3
];
var CreateUsagePlanRequest$ = [3, n0, _CUPR,
    0,
    [_n, _d, _aS, _th, _q, _t],
    [0, 0, () => ListOfApiStage, () => ThrottleSettings$, () => QuotaSettings$, 128 | 0], 1
];
var CreateVpcLinkRequest$ = [3, n0, _CVLR,
    0,
    [_n, _tA, _d, _t],
    [0, 64 | 0, 0, 128 | 0], 2
];
var DeleteApiKeyRequest$ = [3, n0, _DAKR,
    0,
    [_aK],
    [[0, 1]], 1
];
var DeleteAuthorizerRequest$ = [3, n0, _DAR,
    0,
    [_rAI, _aIu],
    [[0, 1], [0, 1]], 2
];
var DeleteBasePathMappingRequest$ = [3, n0, _DBPMR,
    0,
    [_dN, _bP, _dNI],
    [[0, 1], [0, 1], [0, { [_hQ]: _dNI }]], 2
];
var DeleteClientCertificateRequest$ = [3, n0, _DCCR,
    0,
    [_cCI],
    [[0, 1]], 1
];
var DeleteDeploymentRequest$ = [3, n0, _DDR,
    0,
    [_rAI, _dI],
    [[0, 1], [0, 1]], 2
];
var DeleteDocumentationPartRequest$ = [3, n0, _DDPR,
    0,
    [_rAI, _dPI],
    [[0, 1], [0, 1]], 2
];
var DeleteDocumentationVersionRequest$ = [3, n0, _DDVR,
    0,
    [_rAI, _dV],
    [[0, 1], [0, 1]], 2
];
var DeleteDomainNameAccessAssociationRequest$ = [3, n0, _DDNAAR,
    0,
    [_dNAAA],
    [[0, 1]], 1
];
var DeleteDomainNameRequest$ = [3, n0, _DDNR,
    0,
    [_dN, _dNI],
    [[0, 1], [0, { [_hQ]: _dNI }]], 1
];
var DeleteGatewayResponseRequest$ = [3, n0, _DGRR,
    0,
    [_rAI, _rT],
    [[0, 1], [0, 1]], 2
];
var DeleteIntegrationRequest$ = [3, n0, _DIR,
    0,
    [_rAI, _rI, _hM],
    [[0, 1], [0, 1], [0, 1]], 3
];
var DeleteIntegrationResponseRequest$ = [3, n0, _DIRR,
    0,
    [_rAI, _rI, _hM, _sC],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
var DeleteMethodRequest$ = [3, n0, _DMR,
    0,
    [_rAI, _rI, _hM],
    [[0, 1], [0, 1], [0, 1]], 3
];
var DeleteMethodResponseRequest$ = [3, n0, _DMRR,
    0,
    [_rAI, _rI, _hM, _sC],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
var DeleteModelRequest$ = [3, n0, _DMRe,
    0,
    [_rAI, _mN],
    [[0, 1], [0, 1]], 2
];
var DeleteRequestValidatorRequest$ = [3, n0, _DRVR,
    0,
    [_rAI, _rVI],
    [[0, 1], [0, 1]], 2
];
var DeleteResourceRequest$ = [3, n0, _DRR,
    0,
    [_rAI, _rI],
    [[0, 1], [0, 1]], 2
];
var DeleteRestApiRequest$ = [3, n0, _DRAR,
    0,
    [_rAI],
    [[0, 1]], 1
];
var DeleteStageRequest$ = [3, n0, _DSR,
    0,
    [_rAI, _sN],
    [[0, 1], [0, 1]], 2
];
var DeleteUsagePlanKeyRequest$ = [3, n0, _DUPKR,
    0,
    [_uPI, _kI],
    [[0, 1], [0, 1]], 2
];
var DeleteUsagePlanRequest$ = [3, n0, _DUPR,
    0,
    [_uPI],
    [[0, 1]], 1
];
var DeleteVpcLinkRequest$ = [3, n0, _DVLR,
    0,
    [_vLI],
    [[0, 1]], 1
];
var Deployment$ = [3, n0, _D,
    0,
    [_i, _d, _cD, _aSp],
    [0, 0, 4, () => PathToMapOfMethodSnapshot]
];
var DeploymentCanarySettings$ = [3, n0, _DCS,
    0,
    [_pT, _sVO, _uSC],
    [1, 128 | 0, 2]
];
var Deployments$ = [3, n0, _De,
    0,
    [_it, _p],
    [[() => ListOfDeployment, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var DocumentationPart$ = [3, n0, _DP,
    0,
    [_i, _l, _pr],
    [0, () => DocumentationPartLocation$, 0]
];
var DocumentationPartIds$ = [3, n0, _DPI,
    0,
    [_id, _w],
    [64 | 0, 64 | 0]
];
var DocumentationPartLocation$ = [3, n0, _DPL,
    0,
    [_ty, _pa, _me, _sC, _n],
    [0, 0, 0, 0, 0], 1
];
var DocumentationParts$ = [3, n0, _DPo,
    0,
    [_it, _p],
    [[() => ListOfDocumentationPart, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var DocumentationVersion$ = [3, n0, _DV,
    0,
    [_ve, _cD, _d],
    [0, 4, 0]
];
var DocumentationVersions$ = [3, n0, _DVo,
    0,
    [_it, _p],
    [[() => ListOfDocumentationVersion, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var DomainName$ = [3, n0, _DN,
    0,
    [_dN, _dNI, _dNA, _cN, _cA, _cUD, _rDN, _rHZI, _rCN, _rCA, _dDN, _dHZI, _eC, _dNS, _dNSM, _sP, _eAM, _t, _mTA, _oVCA, _mP, _po, _rM],
    [0, 0, 0, 0, 0, 4, 0, 0, 0, 0, 0, 0, () => EndpointConfiguration$, 0, 0, 0, 0, 128 | 0, () => MutualTlsAuthentication$, 0, 0, 0, 0]
];
var DomainNameAccessAssociation$ = [3, n0, _DNAA,
    0,
    [_dNAAA, _dNA, _aAST, _aAS, _t],
    [0, 0, 0, 0, 128 | 0]
];
var DomainNameAccessAssociations$ = [3, n0, _DNAAo,
    0,
    [_it, _p],
    [[() => ListOfDomainNameAccessAssociation, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var DomainNames$ = [3, n0, _DNo,
    0,
    [_it, _p],
    [[() => ListOfDomainName, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var EndpointConfiguration$ = [3, n0, _EC,
    0,
    [_typ, _iAT, _vEI],
    [64 | 0, 0, 64 | 0]
];
var ExportResponse$ = [3, n0, _ER,
    0,
    [_cT, _cDo, _b],
    [[0, { [_hH]: _CT }], [0, { [_hH]: _CD }], [21, 16]]
];
var FlushStageAuthorizersCacheRequest$ = [3, n0, _FSACR,
    0,
    [_rAI, _sN],
    [[0, 1], [0, 1]], 2
];
var FlushStageCacheRequest$ = [3, n0, _FSCR,
    0,
    [_rAI, _sN],
    [[0, 1], [0, 1]], 2
];
var GatewayResponse$ = [3, n0, _GR,
    0,
    [_rT, _sC, _rP, _rTe, _dR],
    [0, 0, 128 | 0, 128 | 0, 2]
];
var GatewayResponses$ = [3, n0, _GRa,
    0,
    [_it, _p],
    [[() => ListOfGatewayResponse, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var GenerateClientCertificateRequest$ = [3, n0, _GCCR,
    0,
    [_d, _t],
    [0, 128 | 0]
];
var GetAccountRequest$ = [3, n0, _GAR,
    0,
    [],
    []
];
var GetApiKeyRequest$ = [3, n0, _GAKR,
    0,
    [_aK, _iV],
    [[0, 1], [2, { [_hQ]: _iV }]], 1
];
var GetApiKeysRequest$ = [3, n0, _GAKRe,
    0,
    [_p, _li, _nQ, _cI, _iVn],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _n }], [0, { [_hQ]: _cI }], [2, { [_hQ]: _iVn }]]
];
var GetAuthorizerRequest$ = [3, n0, _GARe,
    0,
    [_rAI, _aIu],
    [[0, 1], [0, 1]], 2
];
var GetAuthorizersRequest$ = [3, n0, _GARet,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
var GetBasePathMappingRequest$ = [3, n0, _GBPMR,
    0,
    [_dN, _bP, _dNI],
    [[0, 1], [0, 1], [0, { [_hQ]: _dNI }]], 2
];
var GetBasePathMappingsRequest$ = [3, n0, _GBPMRe,
    0,
    [_dN, _dNI, _p, _li],
    [[0, 1], [0, { [_hQ]: _dNI }], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
var GetClientCertificateRequest$ = [3, n0, _GCCRe,
    0,
    [_cCI],
    [[0, 1]], 1
];
var GetClientCertificatesRequest$ = [3, n0, _GCCRet,
    0,
    [_p, _li],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }]]
];
var GetDeploymentRequest$ = [3, n0, _GDR,
    0,
    [_rAI, _dI, _em],
    [[0, 1], [0, 1], [64 | 0, { [_hQ]: _em }]], 2
];
var GetDeploymentsRequest$ = [3, n0, _GDRe,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
var GetDocumentationPartRequest$ = [3, n0, _GDPR,
    0,
    [_rAI, _dPI],
    [[0, 1], [0, 1]], 2
];
var GetDocumentationPartsRequest$ = [3, n0, _GDPRe,
    0,
    [_rAI, _ty, _nQ, _pa, _p, _li, _lS],
    [[0, 1], [0, { [_hQ]: _ty }], [0, { [_hQ]: _n }], [0, { [_hQ]: _pa }], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _lS }]], 1
];
var GetDocumentationVersionRequest$ = [3, n0, _GDVR,
    0,
    [_rAI, _dV],
    [[0, 1], [0, 1]], 2
];
var GetDocumentationVersionsRequest$ = [3, n0, _GDVRe,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
var GetDomainNameAccessAssociationsRequest$ = [3, n0, _GDNAAR,
    0,
    [_p, _li, _rO],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _rO }]]
];
var GetDomainNameRequest$ = [3, n0, _GDNR,
    0,
    [_dN, _dNI],
    [[0, 1], [0, { [_hQ]: _dNI }]], 1
];
var GetDomainNamesRequest$ = [3, n0, _GDNRe,
    0,
    [_p, _li, _rO],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _rO }]]
];
var GetExportRequest$ = [3, n0, _GER,
    0,
    [_rAI, _sN, _eT, _par, _a],
    [[0, 1], [0, 1], [0, 1], [128 | 0, 64], [0, { [_hH]: _Ac }]], 3
];
var GetGatewayResponseRequest$ = [3, n0, _GGRR,
    0,
    [_rAI, _rT],
    [[0, 1], [0, 1]], 2
];
var GetGatewayResponsesRequest$ = [3, n0, _GGRRe,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
var GetIntegrationRequest$ = [3, n0, _GIR,
    0,
    [_rAI, _rI, _hM],
    [[0, 1], [0, 1], [0, 1]], 3
];
var GetIntegrationResponseRequest$ = [3, n0, _GIRR,
    0,
    [_rAI, _rI, _hM, _sC],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
var GetMethodRequest$ = [3, n0, _GMR,
    0,
    [_rAI, _rI, _hM],
    [[0, 1], [0, 1], [0, 1]], 3
];
var GetMethodResponseRequest$ = [3, n0, _GMRR,
    0,
    [_rAI, _rI, _hM, _sC],
    [[0, 1], [0, 1], [0, 1], [0, 1]], 4
];
var GetModelRequest$ = [3, n0, _GMRe,
    0,
    [_rAI, _mN, _fl],
    [[0, 1], [0, 1], [2, { [_hQ]: _fl }]], 2
];
var GetModelsRequest$ = [3, n0, _GMRet,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
var GetModelTemplateRequest$ = [3, n0, _GMTR,
    0,
    [_rAI, _mN],
    [[0, 1], [0, 1]], 2
];
var GetRequestValidatorRequest$ = [3, n0, _GRVR,
    0,
    [_rAI, _rVI],
    [[0, 1], [0, 1]], 2
];
var GetRequestValidatorsRequest$ = [3, n0, _GRVRe,
    0,
    [_rAI, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
var GetResourceRequest$ = [3, n0, _GRR,
    0,
    [_rAI, _rI, _em],
    [[0, 1], [0, 1], [64 | 0, { [_hQ]: _em }]], 2
];
var GetResourcesRequest$ = [3, n0, _GRRe,
    0,
    [_rAI, _p, _li, _em],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [64 | 0, { [_hQ]: _em }]], 1
];
var GetRestApiRequest$ = [3, n0, _GRAR,
    0,
    [_rAI],
    [[0, 1]], 1
];
var GetRestApisRequest$ = [3, n0, _GRARe,
    0,
    [_p, _li],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }]]
];
var GetSdkRequest$ = [3, n0, _GSR,
    0,
    [_rAI, _sN, _sT, _par],
    [[0, 1], [0, 1], [0, 1], [128 | 0, 64]], 3
];
var GetSdkTypeRequest$ = [3, n0, _GSTR,
    0,
    [_i],
    [[0, 1]], 1
];
var GetSdkTypesRequest$ = [3, n0, _GSTRe,
    0,
    [_p, _li],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }]]
];
var GetStageRequest$ = [3, n0, _GSRe,
    0,
    [_rAI, _sN],
    [[0, 1], [0, 1]], 2
];
var GetStagesRequest$ = [3, n0, _GSRet,
    0,
    [_rAI, _dI],
    [[0, 1], [0, { [_hQ]: _dI }]], 1
];
var GetTagsRequest$ = [3, n0, _GTR,
    0,
    [_rA, _p, _li],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 1
];
var GetUsagePlanKeyRequest$ = [3, n0, _GUPKR,
    0,
    [_uPI, _kI],
    [[0, 1], [0, 1]], 2
];
var GetUsagePlanKeysRequest$ = [3, n0, _GUPKRe,
    0,
    [_uPI, _p, _li, _nQ],
    [[0, 1], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }], [0, { [_hQ]: _n }]], 1
];
var GetUsagePlanRequest$ = [3, n0, _GUPR,
    0,
    [_uPI],
    [[0, 1]], 1
];
var GetUsagePlansRequest$ = [3, n0, _GUPRe,
    0,
    [_p, _kI, _li],
    [[0, { [_hQ]: _p }], [0, { [_hQ]: _kI }], [1, { [_hQ]: _li }]]
];
var GetUsageRequest$ = [3, n0, _GUR,
    0,
    [_uPI, _sDt, _eDn, _kI, _p, _li],
    [[0, 1], [0, { [_hQ]: _sDt }], [0, { [_hQ]: _eDn }], [0, { [_hQ]: _kI }], [0, { [_hQ]: _p }], [1, { [_hQ]: _li }]], 3
];
var GetVpcLinkRequest$ = [3, n0, _GVLR,
    0,
    [_vLI],
    [[0, 1]], 1
];
var GetVpcLinksRequest$ = [3, n0, _GVLRe,
    0,
    [_p, _li],
    [[0, { [_hQ]: _p }], [1, { [_hQ]: _li }]]
];
var ImportApiKeysRequest$ = [3, n0, _IAKR,
    0,
    [_b, _f, _fOW],
    [[21, 16], [0, { [_hQ]: _f }], [2, { [_hQ]: _fa }]], 2
];
var ImportDocumentationPartsRequest$ = [3, n0, _IDPR,
    0,
    [_rAI, _b, _mo, _fOW],
    [[0, 1], [21, 16], [0, { [_hQ]: _mo }], [2, { [_hQ]: _fa }]], 2
];
var ImportRestApiRequest$ = [3, n0, _IRAR,
    0,
    [_b, _fOW, _par],
    [[21, 16], [2, { [_hQ]: _fa }], [128 | 0, 64]], 1
];
var Integration$ = [3, n0, _I,
    0,
    [_ty, _hM, _u, _cTo, _cIo, _cr, _rPe, _rTeq, _pB, _cH, _tIM, _cNa, _cKP, _iR, _tC, _rTM, _iT],
    [0, 0, 0, 0, 0, 0, 128 | 0, 128 | 0, 0, 0, 1, 0, 64 | 0, () => MapOfIntegrationResponse, () => TlsConfig$, 0, 0]
];
var IntegrationResponse$ = [3, n0, _IR,
    0,
    [_sC, _sPe, _rP, _rTe, _cH],
    [0, 0, 128 | 0, 128 | 0, 0]
];
var LimitExceededException$ = [-3, n0, _LEE,
    { [_er]: _c, [_hE]: 429 },
    [_rAS, _m],
    [[0, { [_hH]: _RA }], 0]
];
schema.TypeRegistry.for(n0).registerError(LimitExceededException$, LimitExceededException);
var Method$ = [3, n0, _M,
    0,
    [_hM, _aTu, _aIu, _aKR, _rVI, _oN, _rPe, _rMe, _mR, _mI, _aSu],
    [0, 0, 0, 2, 0, 0, 128 | 2, 128 | 0, () => MapOfMethodResponse, () => Integration$, 64 | 0]
];
var MethodResponse$ = [3, n0, _MR,
    0,
    [_sC, _rP, _rMes],
    [0, 128 | 2, 128 | 0]
];
var MethodSetting$ = [3, n0, _MS,
    0,
    [_mE, _lL, _dTE, _tBL, _tRL, _cE, _cTIS, _cDE, _rAFCC, _uCCHS],
    [2, 0, 2, 1, 1, 2, 1, 2, 2, 0]
];
var MethodSnapshot$ = [3, n0, _MSe,
    0,
    [_aTu, _aKR],
    [0, 2]
];
var Model$ = [3, n0, _Mo,
    0,
    [_i, _n, _d, _sc, _cT],
    [0, 0, 0, 0, 0]
];
var Models$ = [3, n0, _Mod,
    0,
    [_it, _p],
    [[() => ListOfModel, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var MutualTlsAuthentication$ = [3, n0, _MTA,
    0,
    [_tU, _tV, _tW],
    [0, 0, 64 | 0]
];
var MutualTlsAuthenticationInput$ = [3, n0, _MTAI,
    0,
    [_tU, _tV],
    [0, 0]
];
var NotFoundException$ = [-3, n0, _NFE,
    { [_er]: _c, [_hE]: 404 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(NotFoundException$, NotFoundException);
var PatchOperation$ = [3, n0, _PO,
    0,
    [_o, _pa, _v, _fr],
    [0, 0, 0, 0]
];
var PutGatewayResponseRequest$ = [3, n0, _PGRR,
    0,
    [_rAI, _rT, _sC, _rP, _rTe],
    [[0, 1], [0, 1], 0, 128 | 0, 128 | 0], 2
];
var PutIntegrationRequest$ = [3, n0, _PIR,
    0,
    [_rAI, _rI, _hM, _ty, _iHM, _u, _cTo, _cIo, _cr, _rPe, _rTeq, _pB, _cNa, _cKP, _cH, _tIM, _tC, _rTM, _iT],
    [[0, 1], [0, 1], [0, { [_hL]: 1, [_jN]: _rHM }], 0, [0, { [_jN]: _hM }], 0, 0, 0, 0, 128 | 0, 128 | 0, 0, 0, 64 | 0, 0, 1, () => TlsConfig$, 0, 0], 4
];
var PutIntegrationResponseRequest$ = [3, n0, _PIRR,
    0,
    [_rAI, _rI, _hM, _sC, _sPe, _rP, _rTe, _cH],
    [[0, 1], [0, 1], [0, 1], [0, 1], 0, 128 | 0, 128 | 0, 0], 4
];
var PutMethodRequest$ = [3, n0, _PMR,
    0,
    [_rAI, _rI, _hM, _aTu, _aIu, _aKR, _oN, _rPe, _rMe, _rVI, _aSu],
    [[0, 1], [0, 1], [0, 1], 0, 0, 2, 0, 128 | 2, 128 | 0, 0, 64 | 0], 4
];
var PutMethodResponseRequest$ = [3, n0, _PMRR,
    0,
    [_rAI, _rI, _hM, _sC, _rP, _rMes],
    [[0, 1], [0, 1], [0, 1], [0, 1], 128 | 2, 128 | 0], 4
];
var PutRestApiRequest$ = [3, n0, _PRAR,
    0,
    [_rAI, _b, _mo, _fOW, _par],
    [[0, 1], [21, 16], [0, { [_hQ]: _mo }], [2, { [_hQ]: _fa }], [128 | 0, 64]], 2
];
var QuotaSettings$ = [3, n0, _QS,
    0,
    [_li, _of, _pe],
    [1, 1, 0]
];
var RejectDomainNameAccessAssociationRequest$ = [3, n0, _RDNAAR,
    0,
    [_dNAAA, _dNA],
    [[0, { [_hQ]: _dNAAA }], [0, { [_hQ]: _dNA }]], 2
];
var RequestValidator$ = [3, n0, _RV,
    0,
    [_i, _n, _vRB, _vRP],
    [0, 0, 2, 2]
];
var RequestValidators$ = [3, n0, _RVe,
    0,
    [_it, _p],
    [[() => ListOfRequestValidator, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var Resource$ = [3, n0, _R,
    0,
    [_i, _pI, _pP, _pa, _rMeso],
    [0, 0, 0, 0, () => MapOfMethod]
];
var Resources$ = [3, n0, _Re,
    0,
    [_it, _p],
    [[() => ListOfResource, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var RestApi$ = [3, n0, _RAe,
    0,
    [_i, _n, _d, _cD, _ve, _w, _bMT, _mCS, _aKS, _eC, _po, _t, _dEAE, _rRI, _sP, _eAM, _aSpi, _aSM],
    [0, 0, 0, 4, 0, 64 | 0, 64 | 0, 1, 0, () => EndpointConfiguration$, 0, 128 | 0, 2, 0, 0, 0, 0, 0]
];
var RestApis$ = [3, n0, _RAes,
    0,
    [_it, _p],
    [[() => ListOfRestApi, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var SdkConfigurationProperty$ = [3, n0, _SCP,
    0,
    [_n, _fN, _d, _r, _dVe],
    [0, 0, 0, 2, 0]
];
var SdkResponse$ = [3, n0, _SR,
    0,
    [_cT, _cDo, _b],
    [[0, { [_hH]: _CT }], [0, { [_hH]: _CD }], [21, 16]]
];
var SdkType$ = [3, n0, _ST,
    0,
    [_i, _fN, _d, _cP],
    [0, 0, 0, () => ListOfSdkConfigurationProperty]
];
var SdkTypes$ = [3, n0, _STd,
    0,
    [_it],
    [[() => ListOfSdkType, { [_jN]: _ite }]]
];
var ServiceUnavailableException$ = [-3, n0, _SUE,
    { [_er]: _se, [_hE]: 503 },
    [_rAS, _m],
    [[0, { [_hH]: _RA }], 0]
];
schema.TypeRegistry.for(n0).registerError(ServiceUnavailableException$, ServiceUnavailableException);
var Stage$ = [3, n0, _S,
    0,
    [_dI, _cCI, _sN, _d, _cCE, _cCS, _cCSa, _mS, _va, _dV, _aLS, _cS, _tE, _wAA, _t, _cD, _lUD],
    [0, 0, 0, 0, 2, 0, 0, () => MapOfMethodSettings, 128 | 0, 0, () => AccessLogSettings$, () => CanarySettings$, 2, 0, 128 | 0, 4, 4]
];
var StageKey$ = [3, n0, _SK,
    0,
    [_rAI, _sN],
    [0, 0]
];
var Stages$ = [3, n0, _St,
    0,
    [_ite],
    [() => ListOfStage]
];
var TagResourceRequest$ = [3, n0, _TRR,
    0,
    [_rA, _t],
    [[0, 1], 128 | 0], 2
];
var Tags$ = [3, n0, _T,
    0,
    [_t],
    [128 | 0]
];
var Template$ = [3, n0, _Te,
    0,
    [_v],
    [0]
];
var TestInvokeAuthorizerRequest$ = [3, n0, _TIAR,
    0,
    [_rAI, _aIu, _h, _mVH, _pWQS, _b, _sV, _aCd],
    [[0, 1], [0, 1], 128 | 0, [2, n0, _MOSTL, 0, 0, 64 | 0], 0, 0, 128 | 0, 128 | 0], 2
];
var TestInvokeAuthorizerResponse$ = [3, n0, _TIARe,
    0,
    [_cSl, _lo, _la, _pIr, _po, _au, _cl],
    [1, 0, 1, 0, 0, [2, n0, _MOSTL, 0, 0, 64 | 0], 128 | 0]
];
var TestInvokeMethodRequest$ = [3, n0, _TIMR,
    0,
    [_rAI, _rI, _hM, _pWQS, _b, _h, _mVH, _cCI, _sV],
    [[0, 1], [0, 1], [0, 1], 0, 0, 128 | 0, [2, n0, _MOSTL, 0, 0, 64 | 0], 0, 128 | 0], 3
];
var TestInvokeMethodResponse$ = [3, n0, _TIMRe,
    0,
    [_st, _b, _h, _mVH, _lo, _la],
    [1, 0, 128 | 0, [2, n0, _MOSTL, 0, 0, 64 | 0], 0, 1]
];
var ThrottleSettings$ = [3, n0, _TS,
    0,
    [_bL, _rL],
    [1, 1]
];
var TlsConfig$ = [3, n0, _TC,
    0,
    [_iSV],
    [2]
];
var TooManyRequestsException$ = [-3, n0, _TMRE,
    { [_er]: _c, [_hE]: 429 },
    [_rAS, _m],
    [[0, { [_hH]: _RA }], 0]
];
schema.TypeRegistry.for(n0).registerError(TooManyRequestsException$, TooManyRequestsException);
var UnauthorizedException$ = [-3, n0, _UE,
    { [_er]: _c, [_hE]: 401 },
    [_m],
    [0]
];
schema.TypeRegistry.for(n0).registerError(UnauthorizedException$, UnauthorizedException);
var UntagResourceRequest$ = [3, n0, _URR,
    0,
    [_rA, _tK],
    [[0, 1], [64 | 0, { [_hQ]: _tK }]], 2
];
var UpdateAccountRequest$ = [3, n0, _UAR,
    0,
    [_pO],
    [() => ListOfPatchOperation]
];
var UpdateApiKeyRequest$ = [3, n0, _UAKR,
    0,
    [_aK, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
var UpdateAuthorizerRequest$ = [3, n0, _UARp,
    0,
    [_rAI, _aIu, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateBasePathMappingRequest$ = [3, n0, _UBPMR,
    0,
    [_dN, _bP, _dNI, _pO],
    [[0, 1], [0, 1], [0, { [_hQ]: _dNI }], () => ListOfPatchOperation], 2
];
var UpdateClientCertificateRequest$ = [3, n0, _UCCR,
    0,
    [_cCI, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
var UpdateDeploymentRequest$ = [3, n0, _UDR,
    0,
    [_rAI, _dI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateDocumentationPartRequest$ = [3, n0, _UDPR,
    0,
    [_rAI, _dPI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateDocumentationVersionRequest$ = [3, n0, _UDVR,
    0,
    [_rAI, _dV, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateDomainNameRequest$ = [3, n0, _UDNR,
    0,
    [_dN, _dNI, _pO],
    [[0, 1], [0, { [_hQ]: _dNI }], () => ListOfPatchOperation], 1
];
var UpdateGatewayResponseRequest$ = [3, n0, _UGRR,
    0,
    [_rAI, _rT, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateIntegrationRequest$ = [3, n0, _UIR,
    0,
    [_rAI, _rI, _hM, _pO],
    [[0, 1], [0, 1], [0, 1], () => ListOfPatchOperation], 3
];
var UpdateIntegrationResponseRequest$ = [3, n0, _UIRR,
    0,
    [_rAI, _rI, _hM, _sC, _pO],
    [[0, 1], [0, 1], [0, 1], [0, 1], () => ListOfPatchOperation], 4
];
var UpdateMethodRequest$ = [3, n0, _UMR,
    0,
    [_rAI, _rI, _hM, _pO],
    [[0, 1], [0, 1], [0, 1], () => ListOfPatchOperation], 3
];
var UpdateMethodResponseRequest$ = [3, n0, _UMRR,
    0,
    [_rAI, _rI, _hM, _sC, _pO],
    [[0, 1], [0, 1], [0, 1], [0, 1], () => ListOfPatchOperation], 4
];
var UpdateModelRequest$ = [3, n0, _UMRp,
    0,
    [_rAI, _mN, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateRequestValidatorRequest$ = [3, n0, _URVR,
    0,
    [_rAI, _rVI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateResourceRequest$ = [3, n0, _URRp,
    0,
    [_rAI, _rI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateRestApiRequest$ = [3, n0, _URAR,
    0,
    [_rAI, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
var UpdateStageRequest$ = [3, n0, _USR,
    0,
    [_rAI, _sN, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateUsagePlanRequest$ = [3, n0, _UUPR,
    0,
    [_uPI, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
var UpdateUsageRequest$ = [3, n0, _UUR,
    0,
    [_uPI, _kI, _pO],
    [[0, 1], [0, 1], () => ListOfPatchOperation], 2
];
var UpdateVpcLinkRequest$ = [3, n0, _UVLR,
    0,
    [_vLI, _pO],
    [[0, 1], () => ListOfPatchOperation], 1
];
var Usage$ = [3, n0, _U,
    0,
    [_uPI, _sDt, _eDn, _it, _p],
    [0, 0, 0, [[2, n0, _MOKU, 0, 0, [1, n0, _LOU, 0, 64 | 1]], { [_jN]: _val }], [0, { [_hQ]: _p }]]
];
var UsagePlan$ = [3, n0, _UP,
    0,
    [_i, _n, _d, _aS, _th, _q, _pC, _t],
    [0, 0, 0, () => ListOfApiStage, () => ThrottleSettings$, () => QuotaSettings$, 0, 128 | 0]
];
var UsagePlanKey$ = [3, n0, _UPK,
    0,
    [_i, _ty, _v, _n],
    [0, 0, 0, 0]
];
var UsagePlanKeys$ = [3, n0, _UPKs,
    0,
    [_it, _p],
    [[() => ListOfUsagePlanKey, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var UsagePlans$ = [3, n0, _UPs,
    0,
    [_it, _p],
    [[() => ListOfUsagePlan, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var VpcLink$ = [3, n0, _VL,
    0,
    [_i, _n, _d, _tA, _st, _sM, _t],
    [0, 0, 0, 64 | 0, 0, 0, 128 | 0]
];
var VpcLinks$ = [3, n0, _VLp,
    0,
    [_it, _p],
    [[() => ListOfVpcLink, { [_jN]: _ite }], [0, { [_hQ]: _p }]]
];
var __Unit = "unit";
var APIGatewayServiceException$ = [-3, _sm, "APIGatewayServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(APIGatewayServiceException$, APIGatewayServiceException);
var ListOfApiKey = [1, n0, _LOAK,
    0, () => ApiKey$
];
var ListOfApiStage = [1, n0, _LOAS,
    0, () => ApiStage$
];
var ListOfAuthorizer = [1, n0, _LOA,
    0, () => Authorizer$
];
var ListOfBasePathMapping = [1, n0, _LOBPM,
    0, () => BasePathMapping$
];
var ListOfClientCertificate = [1, n0, _LOCC,
    0, () => ClientCertificate$
];
var ListOfDeployment = [1, n0, _LOD,
    0, () => Deployment$
];
var ListOfDocumentationPart = [1, n0, _LODP,
    0, () => DocumentationPart$
];
var ListOfDocumentationVersion = [1, n0, _LODV,
    0, () => DocumentationVersion$
];
var ListOfDomainName = [1, n0, _LODN,
    0, () => DomainName$
];
var ListOfDomainNameAccessAssociation = [1, n0, _LODNAA,
    0, () => DomainNameAccessAssociation$
];
var ListOfGatewayResponse = [1, n0, _LOGR,
    0, () => GatewayResponse$
];
var ListOfModel = [1, n0, _LOM,
    0, () => Model$
];
var ListOfPatchOperation = [1, n0, _LOPO,
    0, () => PatchOperation$
];
var ListOfRequestValidator = [1, n0, _LORV,
    0, () => RequestValidator$
];
var ListOfResource = [1, n0, _LOR,
    0, () => Resource$
];
var ListOfRestApi = [1, n0, _LORA,
    0, () => RestApi$
];
var ListOfSdkConfigurationProperty = [1, n0, _LOSCP,
    0, () => SdkConfigurationProperty$
];
var ListOfSdkType = [1, n0, _LOST,
    0, () => SdkType$
];
var ListOfStage = [1, n0, _LOS,
    0, () => Stage$
];
var ListOfStageKeys = [1, n0, _LOSK,
    0, () => StageKey$
];
var ListOfUsagePlan = [1, n0, _LOUP,
    0, () => UsagePlan$
];
var ListOfUsagePlanKey = [1, n0, _LOUPK,
    0, () => UsagePlanKey$
];
var ListOfVpcLink = [1, n0, _LOVL,
    0, () => VpcLink$
];
var MapOfApiStageThrottleSettings = [2, n0, _MOASTS,
    0, 0, () => ThrottleSettings$
];
var MapOfIntegrationResponse = [2, n0, _MOIR,
    0, 0, () => IntegrationResponse$
];
var MapOfMethod = [2, n0, _MOM,
    0, 0, () => Method$
];
var MapOfMethodResponse = [2, n0, _MOMR,
    0, 0, () => MethodResponse$
];
var MapOfMethodSettings = [2, n0, _MOMS,
    0, 0, () => MethodSetting$
];
var MapOfMethodSnapshot = [2, n0, _MOMSa,
    0, 0, () => MethodSnapshot$
];
var PathToMapOfMethodSnapshot = [2, n0, _PTMOMS,
    0, 0, () => MapOfMethodSnapshot
];
var CreateApiKey$ = [9, n0, _CAK,
    { [_ht]: ["POST", "/apikeys", 201] }, () => CreateApiKeyRequest$, () => ApiKey$
];
var CreateAuthorizer$ = [9, n0, _CA,
    { [_ht]: ["POST", "/restapis/{restApiId}/authorizers", 201] }, () => CreateAuthorizerRequest$, () => Authorizer$
];
var CreateBasePathMapping$ = [9, n0, _CBPM,
    { [_ht]: ["POST", "/domainnames/{domainName}/basepathmappings", 201] }, () => CreateBasePathMappingRequest$, () => BasePathMapping$
];
var CreateDeployment$ = [9, n0, _CDr,
    { [_ht]: ["POST", "/restapis/{restApiId}/deployments", 201] }, () => CreateDeploymentRequest$, () => Deployment$
];
var CreateDocumentationPart$ = [9, n0, _CDP,
    { [_ht]: ["POST", "/restapis/{restApiId}/documentation/parts", 201] }, () => CreateDocumentationPartRequest$, () => DocumentationPart$
];
var CreateDocumentationVersion$ = [9, n0, _CDV,
    { [_ht]: ["POST", "/restapis/{restApiId}/documentation/versions", 201] }, () => CreateDocumentationVersionRequest$, () => DocumentationVersion$
];
var CreateDomainName$ = [9, n0, _CDN,
    { [_ht]: ["POST", "/domainnames", 201] }, () => CreateDomainNameRequest$, () => DomainName$
];
var CreateDomainNameAccessAssociation$ = [9, n0, _CDNAA,
    { [_ht]: ["POST", "/domainnameaccessassociations", 201] }, () => CreateDomainNameAccessAssociationRequest$, () => DomainNameAccessAssociation$
];
var CreateModel$ = [9, n0, _CM,
    { [_ht]: ["POST", "/restapis/{restApiId}/models", 201] }, () => CreateModelRequest$, () => Model$
];
var CreateRequestValidator$ = [9, n0, _CRV,
    { [_ht]: ["POST", "/restapis/{restApiId}/requestvalidators", 201] }, () => CreateRequestValidatorRequest$, () => RequestValidator$
];
var CreateResource$ = [9, n0, _CR,
    { [_ht]: ["POST", "/restapis/{restApiId}/resources/{parentId}", 201] }, () => CreateResourceRequest$, () => Resource$
];
var CreateRestApi$ = [9, n0, _CRA,
    { [_ht]: ["POST", "/restapis", 201] }, () => CreateRestApiRequest$, () => RestApi$
];
var CreateStage$ = [9, n0, _CSr,
    { [_ht]: ["POST", "/restapis/{restApiId}/stages", 201] }, () => CreateStageRequest$, () => Stage$
];
var CreateUsagePlan$ = [9, n0, _CUP,
    { [_ht]: ["POST", "/usageplans", 201] }, () => CreateUsagePlanRequest$, () => UsagePlan$
];
var CreateUsagePlanKey$ = [9, n0, _CUPK,
    { [_ht]: ["POST", "/usageplans/{usagePlanId}/keys", 201] }, () => CreateUsagePlanKeyRequest$, () => UsagePlanKey$
];
var CreateVpcLink$ = [9, n0, _CVL,
    { [_ht]: ["POST", "/vpclinks", 202] }, () => CreateVpcLinkRequest$, () => VpcLink$
];
var DeleteApiKey$ = [9, n0, _DAK,
    { [_ht]: ["DELETE", "/apikeys/{apiKey}", 202] }, () => DeleteApiKeyRequest$, () => __Unit
];
var DeleteAuthorizer$ = [9, n0, _DA,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/authorizers/{authorizerId}", 202] }, () => DeleteAuthorizerRequest$, () => __Unit
];
var DeleteBasePathMapping$ = [9, n0, _DBPM,
    { [_ht]: ["DELETE", "/domainnames/{domainName}/basepathmappings/{basePath}", 202] }, () => DeleteBasePathMappingRequest$, () => __Unit
];
var DeleteClientCertificate$ = [9, n0, _DCC,
    { [_ht]: ["DELETE", "/clientcertificates/{clientCertificateId}", 202] }, () => DeleteClientCertificateRequest$, () => __Unit
];
var DeleteDeployment$ = [9, n0, _DD,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/deployments/{deploymentId}", 202] }, () => DeleteDeploymentRequest$, () => __Unit
];
var DeleteDocumentationPart$ = [9, n0, _DDP,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/documentation/parts/{documentationPartId}", 202] }, () => DeleteDocumentationPartRequest$, () => __Unit
];
var DeleteDocumentationVersion$ = [9, n0, _DDV,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/documentation/versions/{documentationVersion}", 202] }, () => DeleteDocumentationVersionRequest$, () => __Unit
];
var DeleteDomainName$ = [9, n0, _DDN,
    { [_ht]: ["DELETE", "/domainnames/{domainName}", 202] }, () => DeleteDomainNameRequest$, () => __Unit
];
var DeleteDomainNameAccessAssociation$ = [9, n0, _DDNAA,
    { [_ht]: ["DELETE", "/domainnameaccessassociations/{domainNameAccessAssociationArn}", 202] }, () => DeleteDomainNameAccessAssociationRequest$, () => __Unit
];
var DeleteGatewayResponse$ = [9, n0, _DGR,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/gatewayresponses/{responseType}", 202] }, () => DeleteGatewayResponseRequest$, () => __Unit
];
var DeleteIntegration$ = [9, n0, _DI,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration", 204] }, () => DeleteIntegrationRequest$, () => __Unit
];
var DeleteIntegrationResponse$ = [9, n0, _DIRe,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration/responses/{statusCode}", 204] }, () => DeleteIntegrationResponseRequest$, () => __Unit
];
var DeleteMethod$ = [9, n0, _DM,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 204] }, () => DeleteMethodRequest$, () => __Unit
];
var DeleteMethodResponse$ = [9, n0, _DMRel,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/responses/{statusCode}", 204] }, () => DeleteMethodResponseRequest$, () => __Unit
];
var DeleteModel$ = [9, n0, _DMe,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/models/{modelName}", 202] }, () => DeleteModelRequest$, () => __Unit
];
var DeleteRequestValidator$ = [9, n0, _DRV,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/requestvalidators/{requestValidatorId}", 202] }, () => DeleteRequestValidatorRequest$, () => __Unit
];
var DeleteResource$ = [9, n0, _DR,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/resources/{resourceId}", 202] }, () => DeleteResourceRequest$, () => __Unit
];
var DeleteRestApi$ = [9, n0, _DRA,
    { [_ht]: ["DELETE", "/restapis/{restApiId}", 202] }, () => DeleteRestApiRequest$, () => __Unit
];
var DeleteStage$ = [9, n0, _DS,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/stages/{stageName}", 202] }, () => DeleteStageRequest$, () => __Unit
];
var DeleteUsagePlan$ = [9, n0, _DUP,
    { [_ht]: ["DELETE", "/usageplans/{usagePlanId}", 202] }, () => DeleteUsagePlanRequest$, () => __Unit
];
var DeleteUsagePlanKey$ = [9, n0, _DUPK,
    { [_ht]: ["DELETE", "/usageplans/{usagePlanId}/keys/{keyId}", 202] }, () => DeleteUsagePlanKeyRequest$, () => __Unit
];
var DeleteVpcLink$ = [9, n0, _DVL,
    { [_ht]: ["DELETE", "/vpclinks/{vpcLinkId}", 202] }, () => DeleteVpcLinkRequest$, () => __Unit
];
var FlushStageAuthorizersCache$ = [9, n0, _FSAC,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/stages/{stageName}/cache/authorizers", 202] }, () => FlushStageAuthorizersCacheRequest$, () => __Unit
];
var FlushStageCache$ = [9, n0, _FSC,
    { [_ht]: ["DELETE", "/restapis/{restApiId}/stages/{stageName}/cache/data", 202] }, () => FlushStageCacheRequest$, () => __Unit
];
var GenerateClientCertificate$ = [9, n0, _GCC,
    { [_ht]: ["POST", "/clientcertificates", 201] }, () => GenerateClientCertificateRequest$, () => ClientCertificate$
];
var GetAccount$ = [9, n0, _GA,
    { [_ht]: ["GET", "/account", 200] }, () => GetAccountRequest$, () => Account$
];
var GetApiKey$ = [9, n0, _GAK,
    { [_ht]: ["GET", "/apikeys/{apiKey}", 200] }, () => GetApiKeyRequest$, () => ApiKey$
];
var GetApiKeys$ = [9, n0, _GAKe,
    { [_ht]: ["GET", "/apikeys", 200] }, () => GetApiKeysRequest$, () => ApiKeys$
];
var GetAuthorizer$ = [9, n0, _GAe,
    { [_ht]: ["GET", "/restapis/{restApiId}/authorizers/{authorizerId}", 200] }, () => GetAuthorizerRequest$, () => Authorizer$
];
var GetAuthorizers$ = [9, n0, _GAet,
    { [_ht]: ["GET", "/restapis/{restApiId}/authorizers", 200] }, () => GetAuthorizersRequest$, () => Authorizers$
];
var GetBasePathMapping$ = [9, n0, _GBPM,
    { [_ht]: ["GET", "/domainnames/{domainName}/basepathmappings/{basePath}", 200] }, () => GetBasePathMappingRequest$, () => BasePathMapping$
];
var GetBasePathMappings$ = [9, n0, _GBPMe,
    { [_ht]: ["GET", "/domainnames/{domainName}/basepathmappings", 200] }, () => GetBasePathMappingsRequest$, () => BasePathMappings$
];
var GetClientCertificate$ = [9, n0, _GCCe,
    { [_ht]: ["GET", "/clientcertificates/{clientCertificateId}", 200] }, () => GetClientCertificateRequest$, () => ClientCertificate$
];
var GetClientCertificates$ = [9, n0, _GCCet,
    { [_ht]: ["GET", "/clientcertificates", 200] }, () => GetClientCertificatesRequest$, () => ClientCertificates$
];
var GetDeployment$ = [9, n0, _GD,
    { [_ht]: ["GET", "/restapis/{restApiId}/deployments/{deploymentId}", 200] }, () => GetDeploymentRequest$, () => Deployment$
];
var GetDeployments$ = [9, n0, _GDe,
    { [_ht]: ["GET", "/restapis/{restApiId}/deployments", 200] }, () => GetDeploymentsRequest$, () => Deployments$
];
var GetDocumentationPart$ = [9, n0, _GDP,
    { [_ht]: ["GET", "/restapis/{restApiId}/documentation/parts/{documentationPartId}", 200] }, () => GetDocumentationPartRequest$, () => DocumentationPart$
];
var GetDocumentationParts$ = [9, n0, _GDPe,
    { [_ht]: ["GET", "/restapis/{restApiId}/documentation/parts", 200] }, () => GetDocumentationPartsRequest$, () => DocumentationParts$
];
var GetDocumentationVersion$ = [9, n0, _GDV,
    { [_ht]: ["GET", "/restapis/{restApiId}/documentation/versions/{documentationVersion}", 200] }, () => GetDocumentationVersionRequest$, () => DocumentationVersion$
];
var GetDocumentationVersions$ = [9, n0, _GDVe,
    { [_ht]: ["GET", "/restapis/{restApiId}/documentation/versions", 200] }, () => GetDocumentationVersionsRequest$, () => DocumentationVersions$
];
var GetDomainName$ = [9, n0, _GDN,
    { [_ht]: ["GET", "/domainnames/{domainName}", 200] }, () => GetDomainNameRequest$, () => DomainName$
];
var GetDomainNameAccessAssociations$ = [9, n0, _GDNAA,
    { [_ht]: ["GET", "/domainnameaccessassociations", 200] }, () => GetDomainNameAccessAssociationsRequest$, () => DomainNameAccessAssociations$
];
var GetDomainNames$ = [9, n0, _GDNe,
    { [_ht]: ["GET", "/domainnames", 200] }, () => GetDomainNamesRequest$, () => DomainNames$
];
var GetExport$ = [9, n0, _GE,
    { [_ht]: ["GET", "/restapis/{restApiId}/stages/{stageName}/exports/{exportType}", 200] }, () => GetExportRequest$, () => ExportResponse$
];
var GetGatewayResponse$ = [9, n0, _GGR,
    { [_ht]: ["GET", "/restapis/{restApiId}/gatewayresponses/{responseType}", 200] }, () => GetGatewayResponseRequest$, () => GatewayResponse$
];
var GetGatewayResponses$ = [9, n0, _GGRe,
    { [_ht]: ["GET", "/restapis/{restApiId}/gatewayresponses", 200] }, () => GetGatewayResponsesRequest$, () => GatewayResponses$
];
var GetIntegration$ = [9, n0, _GI,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration", 200] }, () => GetIntegrationRequest$, () => Integration$
];
var GetIntegrationResponse$ = [9, n0, _GIRe,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration/responses/{statusCode}", 200] }, () => GetIntegrationResponseRequest$, () => IntegrationResponse$
];
var GetMethod$ = [9, n0, _GM,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 200] }, () => GetMethodRequest$, () => Method$
];
var GetMethodResponse$ = [9, n0, _GMRete,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/responses/{statusCode}", 200] }, () => GetMethodResponseRequest$, () => MethodResponse$
];
var GetModel$ = [9, n0, _GMe,
    { [_ht]: ["GET", "/restapis/{restApiId}/models/{modelName}", 200] }, () => GetModelRequest$, () => Model$
];
var GetModels$ = [9, n0, _GMet,
    { [_ht]: ["GET", "/restapis/{restApiId}/models", 200] }, () => GetModelsRequest$, () => Models$
];
var GetModelTemplate$ = [9, n0, _GMT,
    { [_ht]: ["GET", "/restapis/{restApiId}/models/{modelName}/default_template", 200] }, () => GetModelTemplateRequest$, () => Template$
];
var GetRequestValidator$ = [9, n0, _GRV,
    { [_ht]: ["GET", "/restapis/{restApiId}/requestvalidators/{requestValidatorId}", 200] }, () => GetRequestValidatorRequest$, () => RequestValidator$
];
var GetRequestValidators$ = [9, n0, _GRVe,
    { [_ht]: ["GET", "/restapis/{restApiId}/requestvalidators", 200] }, () => GetRequestValidatorsRequest$, () => RequestValidators$
];
var GetResource$ = [9, n0, _GRe,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources/{resourceId}", 200] }, () => GetResourceRequest$, () => Resource$
];
var GetResources$ = [9, n0, _GRet,
    { [_ht]: ["GET", "/restapis/{restApiId}/resources", 200] }, () => GetResourcesRequest$, () => Resources$
];
var GetRestApi$ = [9, n0, _GRA,
    { [_ht]: ["GET", "/restapis/{restApiId}", 200] }, () => GetRestApiRequest$, () => RestApi$
];
var GetRestApis$ = [9, n0, _GRAe,
    { [_ht]: ["GET", "/restapis", 200] }, () => GetRestApisRequest$, () => RestApis$
];
var GetSdk$ = [9, n0, _GS,
    { [_ht]: ["GET", "/restapis/{restApiId}/stages/{stageName}/sdks/{sdkType}", 200] }, () => GetSdkRequest$, () => SdkResponse$
];
var GetSdkType$ = [9, n0, _GST,
    { [_ht]: ["GET", "/sdktypes/{id}", 200] }, () => GetSdkTypeRequest$, () => SdkType$
];
var GetSdkTypes$ = [9, n0, _GSTe,
    { [_ht]: ["GET", "/sdktypes", 200] }, () => GetSdkTypesRequest$, () => SdkTypes$
];
var GetStage$ = [9, n0, _GSe,
    { [_ht]: ["GET", "/restapis/{restApiId}/stages/{stageName}", 200] }, () => GetStageRequest$, () => Stage$
];
var GetStages$ = [9, n0, _GSet,
    { [_ht]: ["GET", "/restapis/{restApiId}/stages", 200] }, () => GetStagesRequest$, () => Stages$
];
var GetTags$ = [9, n0, _GT,
    { [_ht]: ["GET", "/tags/{resourceArn}", 200] }, () => GetTagsRequest$, () => Tags$
];
var GetUsage$ = [9, n0, _GU,
    { [_ht]: ["GET", "/usageplans/{usagePlanId}/usage", 200] }, () => GetUsageRequest$, () => Usage$
];
var GetUsagePlan$ = [9, n0, _GUP,
    { [_ht]: ["GET", "/usageplans/{usagePlanId}", 200] }, () => GetUsagePlanRequest$, () => UsagePlan$
];
var GetUsagePlanKey$ = [9, n0, _GUPK,
    { [_ht]: ["GET", "/usageplans/{usagePlanId}/keys/{keyId}", 200] }, () => GetUsagePlanKeyRequest$, () => UsagePlanKey$
];
var GetUsagePlanKeys$ = [9, n0, _GUPKe,
    { [_ht]: ["GET", "/usageplans/{usagePlanId}/keys", 200] }, () => GetUsagePlanKeysRequest$, () => UsagePlanKeys$
];
var GetUsagePlans$ = [9, n0, _GUPe,
    { [_ht]: ["GET", "/usageplans", 200] }, () => GetUsagePlansRequest$, () => UsagePlans$
];
var GetVpcLink$ = [9, n0, _GVL,
    { [_ht]: ["GET", "/vpclinks/{vpcLinkId}", 200] }, () => GetVpcLinkRequest$, () => VpcLink$
];
var GetVpcLinks$ = [9, n0, _GVLe,
    { [_ht]: ["GET", "/vpclinks", 200] }, () => GetVpcLinksRequest$, () => VpcLinks$
];
var ImportApiKeys$ = [9, n0, _IAK,
    { [_ht]: ["POST", "/apikeys?mode=import", 201] }, () => ImportApiKeysRequest$, () => ApiKeyIds$
];
var ImportDocumentationParts$ = [9, n0, _IDP,
    { [_ht]: ["PUT", "/restapis/{restApiId}/documentation/parts", 200] }, () => ImportDocumentationPartsRequest$, () => DocumentationPartIds$
];
var ImportRestApi$ = [9, n0, _IRA,
    { [_ht]: ["POST", "/restapis?mode=import", 201] }, () => ImportRestApiRequest$, () => RestApi$
];
var PutGatewayResponse$ = [9, n0, _PGR,
    { [_ht]: ["PUT", "/restapis/{restApiId}/gatewayresponses/{responseType}", 201] }, () => PutGatewayResponseRequest$, () => GatewayResponse$
];
var PutIntegration$ = [9, n0, _PI,
    { [_ht]: ["PUT", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration", 201] }, () => PutIntegrationRequest$, () => Integration$
];
var PutIntegrationResponse$ = [9, n0, _PIRu,
    { [_ht]: ["PUT", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration/responses/{statusCode}", 201] }, () => PutIntegrationResponseRequest$, () => IntegrationResponse$
];
var PutMethod$ = [9, n0, _PM,
    { [_ht]: ["PUT", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 201] }, () => PutMethodRequest$, () => Method$
];
var PutMethodResponse$ = [9, n0, _PMRu,
    { [_ht]: ["PUT", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/responses/{statusCode}", 201] }, () => PutMethodResponseRequest$, () => MethodResponse$
];
var PutRestApi$ = [9, n0, _PRA,
    { [_ht]: ["PUT", "/restapis/{restApiId}", 200] }, () => PutRestApiRequest$, () => RestApi$
];
var RejectDomainNameAccessAssociation$ = [9, n0, _RDNAA,
    { [_ht]: ["POST", "/rejectdomainnameaccessassociations", 202] }, () => RejectDomainNameAccessAssociationRequest$, () => __Unit
];
var TagResource$ = [9, n0, _TR,
    { [_ht]: ["PUT", "/tags/{resourceArn}", 204] }, () => TagResourceRequest$, () => __Unit
];
var TestInvokeAuthorizer$ = [9, n0, _TIA,
    { [_ht]: ["POST", "/restapis/{restApiId}/authorizers/{authorizerId}", 200] }, () => TestInvokeAuthorizerRequest$, () => TestInvokeAuthorizerResponse$
];
var TestInvokeMethod$ = [9, n0, _TIM,
    { [_ht]: ["POST", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 200] }, () => TestInvokeMethodRequest$, () => TestInvokeMethodResponse$
];
var UntagResource$ = [9, n0, _UR,
    { [_ht]: ["DELETE", "/tags/{resourceArn}", 204] }, () => UntagResourceRequest$, () => __Unit
];
var UpdateAccount$ = [9, n0, _UA,
    { [_ht]: ["PATCH", "/account", 200] }, () => UpdateAccountRequest$, () => Account$
];
var UpdateApiKey$ = [9, n0, _UAK,
    { [_ht]: ["PATCH", "/apikeys/{apiKey}", 200] }, () => UpdateApiKeyRequest$, () => ApiKey$
];
var UpdateAuthorizer$ = [9, n0, _UAp,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/authorizers/{authorizerId}", 200] }, () => UpdateAuthorizerRequest$, () => Authorizer$
];
var UpdateBasePathMapping$ = [9, n0, _UBPM,
    { [_ht]: ["PATCH", "/domainnames/{domainName}/basepathmappings/{basePath}", 200] }, () => UpdateBasePathMappingRequest$, () => BasePathMapping$
];
var UpdateClientCertificate$ = [9, n0, _UCC,
    { [_ht]: ["PATCH", "/clientcertificates/{clientCertificateId}", 200] }, () => UpdateClientCertificateRequest$, () => ClientCertificate$
];
var UpdateDeployment$ = [9, n0, _UD,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/deployments/{deploymentId}", 200] }, () => UpdateDeploymentRequest$, () => Deployment$
];
var UpdateDocumentationPart$ = [9, n0, _UDP,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/documentation/parts/{documentationPartId}", 200] }, () => UpdateDocumentationPartRequest$, () => DocumentationPart$
];
var UpdateDocumentationVersion$ = [9, n0, _UDV,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/documentation/versions/{documentationVersion}", 200] }, () => UpdateDocumentationVersionRequest$, () => DocumentationVersion$
];
var UpdateDomainName$ = [9, n0, _UDN,
    { [_ht]: ["PATCH", "/domainnames/{domainName}", 200] }, () => UpdateDomainNameRequest$, () => DomainName$
];
var UpdateGatewayResponse$ = [9, n0, _UGR,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/gatewayresponses/{responseType}", 200] }, () => UpdateGatewayResponseRequest$, () => GatewayResponse$
];
var UpdateIntegration$ = [9, n0, _UI,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration", 200] }, () => UpdateIntegrationRequest$, () => Integration$
];
var UpdateIntegrationResponse$ = [9, n0, _UIRp,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/integration/responses/{statusCode}", 200] }, () => UpdateIntegrationResponseRequest$, () => IntegrationResponse$
];
var UpdateMethod$ = [9, n0, _UM,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}", 200] }, () => UpdateMethodRequest$, () => Method$
];
var UpdateMethodResponse$ = [9, n0, _UMRpd,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}/methods/{httpMethod}/responses/{statusCode}", 201] }, () => UpdateMethodResponseRequest$, () => MethodResponse$
];
var UpdateModel$ = [9, n0, _UMp,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/models/{modelName}", 200] }, () => UpdateModelRequest$, () => Model$
];
var UpdateRequestValidator$ = [9, n0, _URV,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/requestvalidators/{requestValidatorId}", 200] }, () => UpdateRequestValidatorRequest$, () => RequestValidator$
];
var UpdateResource$ = [9, n0, _URp,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/resources/{resourceId}", 200] }, () => UpdateResourceRequest$, () => Resource$
];
var UpdateRestApi$ = [9, n0, _URA,
    { [_ht]: ["PATCH", "/restapis/{restApiId}", 200] }, () => UpdateRestApiRequest$, () => RestApi$
];
var UpdateStage$ = [9, n0, _US,
    { [_ht]: ["PATCH", "/restapis/{restApiId}/stages/{stageName}", 200] }, () => UpdateStageRequest$, () => Stage$
];
var UpdateUsage$ = [9, n0, _UU,
    { [_ht]: ["PATCH", "/usageplans/{usagePlanId}/keys/{keyId}/usage", 200] }, () => UpdateUsageRequest$, () => Usage$
];
var UpdateUsagePlan$ = [9, n0, _UUP,
    { [_ht]: ["PATCH", "/usageplans/{usagePlanId}", 200] }, () => UpdateUsagePlanRequest$, () => UsagePlan$
];
var UpdateVpcLink$ = [9, n0, _UVL,
    { [_ht]: ["PATCH", "/vpclinks/{vpcLinkId}", 200] }, () => UpdateVpcLinkRequest$, () => VpcLink$
];

class CreateApiKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateApiKey", {})
    .n("APIGatewayClient", "CreateApiKeyCommand")
    .sc(CreateApiKey$)
    .build() {
}

class CreateAuthorizerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateAuthorizer", {})
    .n("APIGatewayClient", "CreateAuthorizerCommand")
    .sc(CreateAuthorizer$)
    .build() {
}

class CreateBasePathMappingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateBasePathMapping", {})
    .n("APIGatewayClient", "CreateBasePathMappingCommand")
    .sc(CreateBasePathMapping$)
    .build() {
}

class CreateDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDeployment", {})
    .n("APIGatewayClient", "CreateDeploymentCommand")
    .sc(CreateDeployment$)
    .build() {
}

class CreateDocumentationPartCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDocumentationPart", {})
    .n("APIGatewayClient", "CreateDocumentationPartCommand")
    .sc(CreateDocumentationPart$)
    .build() {
}

class CreateDocumentationVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDocumentationVersion", {})
    .n("APIGatewayClient", "CreateDocumentationVersionCommand")
    .sc(CreateDocumentationVersion$)
    .build() {
}

class CreateDomainNameAccessAssociationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDomainNameAccessAssociation", {})
    .n("APIGatewayClient", "CreateDomainNameAccessAssociationCommand")
    .sc(CreateDomainNameAccessAssociation$)
    .build() {
}

class CreateDomainNameCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateDomainName", {})
    .n("APIGatewayClient", "CreateDomainNameCommand")
    .sc(CreateDomainName$)
    .build() {
}

class CreateModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateModel", {})
    .n("APIGatewayClient", "CreateModelCommand")
    .sc(CreateModel$)
    .build() {
}

class CreateRequestValidatorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateRequestValidator", {})
    .n("APIGatewayClient", "CreateRequestValidatorCommand")
    .sc(CreateRequestValidator$)
    .build() {
}

class CreateResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateResource", {})
    .n("APIGatewayClient", "CreateResourceCommand")
    .sc(CreateResource$)
    .build() {
}

class CreateRestApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateRestApi", {})
    .n("APIGatewayClient", "CreateRestApiCommand")
    .sc(CreateRestApi$)
    .build() {
}

class CreateStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateStage", {})
    .n("APIGatewayClient", "CreateStageCommand")
    .sc(CreateStage$)
    .build() {
}

class CreateUsagePlanCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateUsagePlan", {})
    .n("APIGatewayClient", "CreateUsagePlanCommand")
    .sc(CreateUsagePlan$)
    .build() {
}

class CreateUsagePlanKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateUsagePlanKey", {})
    .n("APIGatewayClient", "CreateUsagePlanKeyCommand")
    .sc(CreateUsagePlanKey$)
    .build() {
}

class CreateVpcLinkCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "CreateVpcLink", {})
    .n("APIGatewayClient", "CreateVpcLinkCommand")
    .sc(CreateVpcLink$)
    .build() {
}

class DeleteApiKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteApiKey", {})
    .n("APIGatewayClient", "DeleteApiKeyCommand")
    .sc(DeleteApiKey$)
    .build() {
}

class DeleteAuthorizerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteAuthorizer", {})
    .n("APIGatewayClient", "DeleteAuthorizerCommand")
    .sc(DeleteAuthorizer$)
    .build() {
}

class DeleteBasePathMappingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteBasePathMapping", {})
    .n("APIGatewayClient", "DeleteBasePathMappingCommand")
    .sc(DeleteBasePathMapping$)
    .build() {
}

class DeleteClientCertificateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteClientCertificate", {})
    .n("APIGatewayClient", "DeleteClientCertificateCommand")
    .sc(DeleteClientCertificate$)
    .build() {
}

class DeleteDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDeployment", {})
    .n("APIGatewayClient", "DeleteDeploymentCommand")
    .sc(DeleteDeployment$)
    .build() {
}

class DeleteDocumentationPartCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDocumentationPart", {})
    .n("APIGatewayClient", "DeleteDocumentationPartCommand")
    .sc(DeleteDocumentationPart$)
    .build() {
}

class DeleteDocumentationVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDocumentationVersion", {})
    .n("APIGatewayClient", "DeleteDocumentationVersionCommand")
    .sc(DeleteDocumentationVersion$)
    .build() {
}

class DeleteDomainNameAccessAssociationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDomainNameAccessAssociation", {})
    .n("APIGatewayClient", "DeleteDomainNameAccessAssociationCommand")
    .sc(DeleteDomainNameAccessAssociation$)
    .build() {
}

class DeleteDomainNameCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteDomainName", {})
    .n("APIGatewayClient", "DeleteDomainNameCommand")
    .sc(DeleteDomainName$)
    .build() {
}

class DeleteGatewayResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteGatewayResponse", {})
    .n("APIGatewayClient", "DeleteGatewayResponseCommand")
    .sc(DeleteGatewayResponse$)
    .build() {
}

class DeleteIntegrationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteIntegration", {})
    .n("APIGatewayClient", "DeleteIntegrationCommand")
    .sc(DeleteIntegration$)
    .build() {
}

class DeleteIntegrationResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteIntegrationResponse", {})
    .n("APIGatewayClient", "DeleteIntegrationResponseCommand")
    .sc(DeleteIntegrationResponse$)
    .build() {
}

class DeleteMethodCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteMethod", {})
    .n("APIGatewayClient", "DeleteMethodCommand")
    .sc(DeleteMethod$)
    .build() {
}

class DeleteMethodResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteMethodResponse", {})
    .n("APIGatewayClient", "DeleteMethodResponseCommand")
    .sc(DeleteMethodResponse$)
    .build() {
}

class DeleteModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteModel", {})
    .n("APIGatewayClient", "DeleteModelCommand")
    .sc(DeleteModel$)
    .build() {
}

class DeleteRequestValidatorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteRequestValidator", {})
    .n("APIGatewayClient", "DeleteRequestValidatorCommand")
    .sc(DeleteRequestValidator$)
    .build() {
}

class DeleteResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteResource", {})
    .n("APIGatewayClient", "DeleteResourceCommand")
    .sc(DeleteResource$)
    .build() {
}

class DeleteRestApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteRestApi", {})
    .n("APIGatewayClient", "DeleteRestApiCommand")
    .sc(DeleteRestApi$)
    .build() {
}

class DeleteStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteStage", {})
    .n("APIGatewayClient", "DeleteStageCommand")
    .sc(DeleteStage$)
    .build() {
}

class DeleteUsagePlanCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteUsagePlan", {})
    .n("APIGatewayClient", "DeleteUsagePlanCommand")
    .sc(DeleteUsagePlan$)
    .build() {
}

class DeleteUsagePlanKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteUsagePlanKey", {})
    .n("APIGatewayClient", "DeleteUsagePlanKeyCommand")
    .sc(DeleteUsagePlanKey$)
    .build() {
}

class DeleteVpcLinkCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "DeleteVpcLink", {})
    .n("APIGatewayClient", "DeleteVpcLinkCommand")
    .sc(DeleteVpcLink$)
    .build() {
}

class FlushStageAuthorizersCacheCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "FlushStageAuthorizersCache", {})
    .n("APIGatewayClient", "FlushStageAuthorizersCacheCommand")
    .sc(FlushStageAuthorizersCache$)
    .build() {
}

class FlushStageCacheCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "FlushStageCache", {})
    .n("APIGatewayClient", "FlushStageCacheCommand")
    .sc(FlushStageCache$)
    .build() {
}

class GenerateClientCertificateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GenerateClientCertificate", {})
    .n("APIGatewayClient", "GenerateClientCertificateCommand")
    .sc(GenerateClientCertificate$)
    .build() {
}

class GetAccountCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetAccount", {})
    .n("APIGatewayClient", "GetAccountCommand")
    .sc(GetAccount$)
    .build() {
}

class GetApiKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetApiKey", {})
    .n("APIGatewayClient", "GetApiKeyCommand")
    .sc(GetApiKey$)
    .build() {
}

class GetApiKeysCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetApiKeys", {})
    .n("APIGatewayClient", "GetApiKeysCommand")
    .sc(GetApiKeys$)
    .build() {
}

class GetAuthorizerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetAuthorizer", {})
    .n("APIGatewayClient", "GetAuthorizerCommand")
    .sc(GetAuthorizer$)
    .build() {
}

class GetAuthorizersCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetAuthorizers", {})
    .n("APIGatewayClient", "GetAuthorizersCommand")
    .sc(GetAuthorizers$)
    .build() {
}

class GetBasePathMappingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetBasePathMapping", {})
    .n("APIGatewayClient", "GetBasePathMappingCommand")
    .sc(GetBasePathMapping$)
    .build() {
}

class GetBasePathMappingsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetBasePathMappings", {})
    .n("APIGatewayClient", "GetBasePathMappingsCommand")
    .sc(GetBasePathMappings$)
    .build() {
}

class GetClientCertificateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetClientCertificate", {})
    .n("APIGatewayClient", "GetClientCertificateCommand")
    .sc(GetClientCertificate$)
    .build() {
}

class GetClientCertificatesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetClientCertificates", {})
    .n("APIGatewayClient", "GetClientCertificatesCommand")
    .sc(GetClientCertificates$)
    .build() {
}

class GetDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDeployment", {})
    .n("APIGatewayClient", "GetDeploymentCommand")
    .sc(GetDeployment$)
    .build() {
}

class GetDeploymentsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDeployments", {})
    .n("APIGatewayClient", "GetDeploymentsCommand")
    .sc(GetDeployments$)
    .build() {
}

class GetDocumentationPartCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDocumentationPart", {})
    .n("APIGatewayClient", "GetDocumentationPartCommand")
    .sc(GetDocumentationPart$)
    .build() {
}

class GetDocumentationPartsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDocumentationParts", {})
    .n("APIGatewayClient", "GetDocumentationPartsCommand")
    .sc(GetDocumentationParts$)
    .build() {
}

class GetDocumentationVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDocumentationVersion", {})
    .n("APIGatewayClient", "GetDocumentationVersionCommand")
    .sc(GetDocumentationVersion$)
    .build() {
}

class GetDocumentationVersionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDocumentationVersions", {})
    .n("APIGatewayClient", "GetDocumentationVersionsCommand")
    .sc(GetDocumentationVersions$)
    .build() {
}

class GetDomainNameAccessAssociationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDomainNameAccessAssociations", {})
    .n("APIGatewayClient", "GetDomainNameAccessAssociationsCommand")
    .sc(GetDomainNameAccessAssociations$)
    .build() {
}

class GetDomainNameCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDomainName", {})
    .n("APIGatewayClient", "GetDomainNameCommand")
    .sc(GetDomainName$)
    .build() {
}

class GetDomainNamesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetDomainNames", {})
    .n("APIGatewayClient", "GetDomainNamesCommand")
    .sc(GetDomainNames$)
    .build() {
}

class GetExportCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetExport", {})
    .n("APIGatewayClient", "GetExportCommand")
    .sc(GetExport$)
    .build() {
}

class GetGatewayResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetGatewayResponse", {})
    .n("APIGatewayClient", "GetGatewayResponseCommand")
    .sc(GetGatewayResponse$)
    .build() {
}

class GetGatewayResponsesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetGatewayResponses", {})
    .n("APIGatewayClient", "GetGatewayResponsesCommand")
    .sc(GetGatewayResponses$)
    .build() {
}

class GetIntegrationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetIntegration", {})
    .n("APIGatewayClient", "GetIntegrationCommand")
    .sc(GetIntegration$)
    .build() {
}

class GetIntegrationResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetIntegrationResponse", {})
    .n("APIGatewayClient", "GetIntegrationResponseCommand")
    .sc(GetIntegrationResponse$)
    .build() {
}

class GetMethodCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetMethod", {})
    .n("APIGatewayClient", "GetMethodCommand")
    .sc(GetMethod$)
    .build() {
}

class GetMethodResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetMethodResponse", {})
    .n("APIGatewayClient", "GetMethodResponseCommand")
    .sc(GetMethodResponse$)
    .build() {
}

class GetModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetModel", {})
    .n("APIGatewayClient", "GetModelCommand")
    .sc(GetModel$)
    .build() {
}

class GetModelsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetModels", {})
    .n("APIGatewayClient", "GetModelsCommand")
    .sc(GetModels$)
    .build() {
}

class GetModelTemplateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetModelTemplate", {})
    .n("APIGatewayClient", "GetModelTemplateCommand")
    .sc(GetModelTemplate$)
    .build() {
}

class GetRequestValidatorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetRequestValidator", {})
    .n("APIGatewayClient", "GetRequestValidatorCommand")
    .sc(GetRequestValidator$)
    .build() {
}

class GetRequestValidatorsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetRequestValidators", {})
    .n("APIGatewayClient", "GetRequestValidatorsCommand")
    .sc(GetRequestValidators$)
    .build() {
}

class GetResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetResource", {})
    .n("APIGatewayClient", "GetResourceCommand")
    .sc(GetResource$)
    .build() {
}

class GetResourcesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetResources", {})
    .n("APIGatewayClient", "GetResourcesCommand")
    .sc(GetResources$)
    .build() {
}

class GetRestApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetRestApi", {})
    .n("APIGatewayClient", "GetRestApiCommand")
    .sc(GetRestApi$)
    .build() {
}

class GetRestApisCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetRestApis", {})
    .n("APIGatewayClient", "GetRestApisCommand")
    .sc(GetRestApis$)
    .build() {
}

class GetSdkCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetSdk", {})
    .n("APIGatewayClient", "GetSdkCommand")
    .sc(GetSdk$)
    .build() {
}

class GetSdkTypeCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetSdkType", {})
    .n("APIGatewayClient", "GetSdkTypeCommand")
    .sc(GetSdkType$)
    .build() {
}

class GetSdkTypesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetSdkTypes", {})
    .n("APIGatewayClient", "GetSdkTypesCommand")
    .sc(GetSdkTypes$)
    .build() {
}

class GetStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetStage", {})
    .n("APIGatewayClient", "GetStageCommand")
    .sc(GetStage$)
    .build() {
}

class GetStagesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetStages", {})
    .n("APIGatewayClient", "GetStagesCommand")
    .sc(GetStages$)
    .build() {
}

class GetTagsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetTags", {})
    .n("APIGatewayClient", "GetTagsCommand")
    .sc(GetTags$)
    .build() {
}

class GetUsageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsage", {})
    .n("APIGatewayClient", "GetUsageCommand")
    .sc(GetUsage$)
    .build() {
}

class GetUsagePlanCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsagePlan", {})
    .n("APIGatewayClient", "GetUsagePlanCommand")
    .sc(GetUsagePlan$)
    .build() {
}

class GetUsagePlanKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsagePlanKey", {})
    .n("APIGatewayClient", "GetUsagePlanKeyCommand")
    .sc(GetUsagePlanKey$)
    .build() {
}

class GetUsagePlanKeysCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsagePlanKeys", {})
    .n("APIGatewayClient", "GetUsagePlanKeysCommand")
    .sc(GetUsagePlanKeys$)
    .build() {
}

class GetUsagePlansCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetUsagePlans", {})
    .n("APIGatewayClient", "GetUsagePlansCommand")
    .sc(GetUsagePlans$)
    .build() {
}

class GetVpcLinkCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetVpcLink", {})
    .n("APIGatewayClient", "GetVpcLinkCommand")
    .sc(GetVpcLink$)
    .build() {
}

class GetVpcLinksCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "GetVpcLinks", {})
    .n("APIGatewayClient", "GetVpcLinksCommand")
    .sc(GetVpcLinks$)
    .build() {
}

class ImportApiKeysCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "ImportApiKeys", {})
    .n("APIGatewayClient", "ImportApiKeysCommand")
    .sc(ImportApiKeys$)
    .build() {
}

class ImportDocumentationPartsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "ImportDocumentationParts", {})
    .n("APIGatewayClient", "ImportDocumentationPartsCommand")
    .sc(ImportDocumentationParts$)
    .build() {
}

class ImportRestApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "ImportRestApi", {})
    .n("APIGatewayClient", "ImportRestApiCommand")
    .sc(ImportRestApi$)
    .build() {
}

class PutGatewayResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutGatewayResponse", {})
    .n("APIGatewayClient", "PutGatewayResponseCommand")
    .sc(PutGatewayResponse$)
    .build() {
}

class PutIntegrationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutIntegration", {})
    .n("APIGatewayClient", "PutIntegrationCommand")
    .sc(PutIntegration$)
    .build() {
}

class PutIntegrationResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutIntegrationResponse", {})
    .n("APIGatewayClient", "PutIntegrationResponseCommand")
    .sc(PutIntegrationResponse$)
    .build() {
}

class PutMethodCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutMethod", {})
    .n("APIGatewayClient", "PutMethodCommand")
    .sc(PutMethod$)
    .build() {
}

class PutMethodResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutMethodResponse", {})
    .n("APIGatewayClient", "PutMethodResponseCommand")
    .sc(PutMethodResponse$)
    .build() {
}

class PutRestApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "PutRestApi", {})
    .n("APIGatewayClient", "PutRestApiCommand")
    .sc(PutRestApi$)
    .build() {
}

class RejectDomainNameAccessAssociationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "RejectDomainNameAccessAssociation", {})
    .n("APIGatewayClient", "RejectDomainNameAccessAssociationCommand")
    .sc(RejectDomainNameAccessAssociation$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "TagResource", {})
    .n("APIGatewayClient", "TagResourceCommand")
    .sc(TagResource$)
    .build() {
}

class TestInvokeAuthorizerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "TestInvokeAuthorizer", {})
    .n("APIGatewayClient", "TestInvokeAuthorizerCommand")
    .sc(TestInvokeAuthorizer$)
    .build() {
}

class TestInvokeMethodCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "TestInvokeMethod", {})
    .n("APIGatewayClient", "TestInvokeMethodCommand")
    .sc(TestInvokeMethod$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UntagResource", {})
    .n("APIGatewayClient", "UntagResourceCommand")
    .sc(UntagResource$)
    .build() {
}

class UpdateAccountCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateAccount", {})
    .n("APIGatewayClient", "UpdateAccountCommand")
    .sc(UpdateAccount$)
    .build() {
}

class UpdateApiKeyCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateApiKey", {})
    .n("APIGatewayClient", "UpdateApiKeyCommand")
    .sc(UpdateApiKey$)
    .build() {
}

class UpdateAuthorizerCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateAuthorizer", {})
    .n("APIGatewayClient", "UpdateAuthorizerCommand")
    .sc(UpdateAuthorizer$)
    .build() {
}

class UpdateBasePathMappingCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateBasePathMapping", {})
    .n("APIGatewayClient", "UpdateBasePathMappingCommand")
    .sc(UpdateBasePathMapping$)
    .build() {
}

class UpdateClientCertificateCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateClientCertificate", {})
    .n("APIGatewayClient", "UpdateClientCertificateCommand")
    .sc(UpdateClientCertificate$)
    .build() {
}

class UpdateDeploymentCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateDeployment", {})
    .n("APIGatewayClient", "UpdateDeploymentCommand")
    .sc(UpdateDeployment$)
    .build() {
}

class UpdateDocumentationPartCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateDocumentationPart", {})
    .n("APIGatewayClient", "UpdateDocumentationPartCommand")
    .sc(UpdateDocumentationPart$)
    .build() {
}

class UpdateDocumentationVersionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateDocumentationVersion", {})
    .n("APIGatewayClient", "UpdateDocumentationVersionCommand")
    .sc(UpdateDocumentationVersion$)
    .build() {
}

class UpdateDomainNameCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateDomainName", {})
    .n("APIGatewayClient", "UpdateDomainNameCommand")
    .sc(UpdateDomainName$)
    .build() {
}

class UpdateGatewayResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateGatewayResponse", {})
    .n("APIGatewayClient", "UpdateGatewayResponseCommand")
    .sc(UpdateGatewayResponse$)
    .build() {
}

class UpdateIntegrationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateIntegration", {})
    .n("APIGatewayClient", "UpdateIntegrationCommand")
    .sc(UpdateIntegration$)
    .build() {
}

class UpdateIntegrationResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateIntegrationResponse", {})
    .n("APIGatewayClient", "UpdateIntegrationResponseCommand")
    .sc(UpdateIntegrationResponse$)
    .build() {
}

class UpdateMethodCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateMethod", {})
    .n("APIGatewayClient", "UpdateMethodCommand")
    .sc(UpdateMethod$)
    .build() {
}

class UpdateMethodResponseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateMethodResponse", {})
    .n("APIGatewayClient", "UpdateMethodResponseCommand")
    .sc(UpdateMethodResponse$)
    .build() {
}

class UpdateModelCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateModel", {})
    .n("APIGatewayClient", "UpdateModelCommand")
    .sc(UpdateModel$)
    .build() {
}

class UpdateRequestValidatorCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateRequestValidator", {})
    .n("APIGatewayClient", "UpdateRequestValidatorCommand")
    .sc(UpdateRequestValidator$)
    .build() {
}

class UpdateResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateResource", {})
    .n("APIGatewayClient", "UpdateResourceCommand")
    .sc(UpdateResource$)
    .build() {
}

class UpdateRestApiCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateRestApi", {})
    .n("APIGatewayClient", "UpdateRestApiCommand")
    .sc(UpdateRestApi$)
    .build() {
}

class UpdateStageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateStage", {})
    .n("APIGatewayClient", "UpdateStageCommand")
    .sc(UpdateStage$)
    .build() {
}

class UpdateUsageCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateUsage", {})
    .n("APIGatewayClient", "UpdateUsageCommand")
    .sc(UpdateUsage$)
    .build() {
}

class UpdateUsagePlanCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateUsagePlan", {})
    .n("APIGatewayClient", "UpdateUsagePlanCommand")
    .sc(UpdateUsagePlan$)
    .build() {
}

class UpdateVpcLinkCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("BackplaneControlService", "UpdateVpcLink", {})
    .n("APIGatewayClient", "UpdateVpcLinkCommand")
    .sc(UpdateVpcLink$)
    .build() {
}

const commands = {
    CreateApiKeyCommand,
    CreateAuthorizerCommand,
    CreateBasePathMappingCommand,
    CreateDeploymentCommand,
    CreateDocumentationPartCommand,
    CreateDocumentationVersionCommand,
    CreateDomainNameCommand,
    CreateDomainNameAccessAssociationCommand,
    CreateModelCommand,
    CreateRequestValidatorCommand,
    CreateResourceCommand,
    CreateRestApiCommand,
    CreateStageCommand,
    CreateUsagePlanCommand,
    CreateUsagePlanKeyCommand,
    CreateVpcLinkCommand,
    DeleteApiKeyCommand,
    DeleteAuthorizerCommand,
    DeleteBasePathMappingCommand,
    DeleteClientCertificateCommand,
    DeleteDeploymentCommand,
    DeleteDocumentationPartCommand,
    DeleteDocumentationVersionCommand,
    DeleteDomainNameCommand,
    DeleteDomainNameAccessAssociationCommand,
    DeleteGatewayResponseCommand,
    DeleteIntegrationCommand,
    DeleteIntegrationResponseCommand,
    DeleteMethodCommand,
    DeleteMethodResponseCommand,
    DeleteModelCommand,
    DeleteRequestValidatorCommand,
    DeleteResourceCommand,
    DeleteRestApiCommand,
    DeleteStageCommand,
    DeleteUsagePlanCommand,
    DeleteUsagePlanKeyCommand,
    DeleteVpcLinkCommand,
    FlushStageAuthorizersCacheCommand,
    FlushStageCacheCommand,
    GenerateClientCertificateCommand,
    GetAccountCommand,
    GetApiKeyCommand,
    GetApiKeysCommand,
    GetAuthorizerCommand,
    GetAuthorizersCommand,
    GetBasePathMappingCommand,
    GetBasePathMappingsCommand,
    GetClientCertificateCommand,
    GetClientCertificatesCommand,
    GetDeploymentCommand,
    GetDeploymentsCommand,
    GetDocumentationPartCommand,
    GetDocumentationPartsCommand,
    GetDocumentationVersionCommand,
    GetDocumentationVersionsCommand,
    GetDomainNameCommand,
    GetDomainNameAccessAssociationsCommand,
    GetDomainNamesCommand,
    GetExportCommand,
    GetGatewayResponseCommand,
    GetGatewayResponsesCommand,
    GetIntegrationCommand,
    GetIntegrationResponseCommand,
    GetMethodCommand,
    GetMethodResponseCommand,
    GetModelCommand,
    GetModelsCommand,
    GetModelTemplateCommand,
    GetRequestValidatorCommand,
    GetRequestValidatorsCommand,
    GetResourceCommand,
    GetResourcesCommand,
    GetRestApiCommand,
    GetRestApisCommand,
    GetSdkCommand,
    GetSdkTypeCommand,
    GetSdkTypesCommand,
    GetStageCommand,
    GetStagesCommand,
    GetTagsCommand,
    GetUsageCommand,
    GetUsagePlanCommand,
    GetUsagePlanKeyCommand,
    GetUsagePlanKeysCommand,
    GetUsagePlansCommand,
    GetVpcLinkCommand,
    GetVpcLinksCommand,
    ImportApiKeysCommand,
    ImportDocumentationPartsCommand,
    ImportRestApiCommand,
    PutGatewayResponseCommand,
    PutIntegrationCommand,
    PutIntegrationResponseCommand,
    PutMethodCommand,
    PutMethodResponseCommand,
    PutRestApiCommand,
    RejectDomainNameAccessAssociationCommand,
    TagResourceCommand,
    TestInvokeAuthorizerCommand,
    TestInvokeMethodCommand,
    UntagResourceCommand,
    UpdateAccountCommand,
    UpdateApiKeyCommand,
    UpdateAuthorizerCommand,
    UpdateBasePathMappingCommand,
    UpdateClientCertificateCommand,
    UpdateDeploymentCommand,
    UpdateDocumentationPartCommand,
    UpdateDocumentationVersionCommand,
    UpdateDomainNameCommand,
    UpdateGatewayResponseCommand,
    UpdateIntegrationCommand,
    UpdateIntegrationResponseCommand,
    UpdateMethodCommand,
    UpdateMethodResponseCommand,
    UpdateModelCommand,
    UpdateRequestValidatorCommand,
    UpdateResourceCommand,
    UpdateRestApiCommand,
    UpdateStageCommand,
    UpdateUsageCommand,
    UpdateUsagePlanCommand,
    UpdateVpcLinkCommand,
};
class APIGateway extends APIGatewayClient {
}
smithyClient.createAggregatedClient(commands, APIGateway);

const paginateGetApiKeys = core.createPaginator(APIGatewayClient, GetApiKeysCommand, "position", "position", "limit");

const paginateGetBasePathMappings = core.createPaginator(APIGatewayClient, GetBasePathMappingsCommand, "position", "position", "limit");

const paginateGetClientCertificates = core.createPaginator(APIGatewayClient, GetClientCertificatesCommand, "position", "position", "limit");

const paginateGetDeployments = core.createPaginator(APIGatewayClient, GetDeploymentsCommand, "position", "position", "limit");

const paginateGetDomainNames = core.createPaginator(APIGatewayClient, GetDomainNamesCommand, "position", "position", "limit");

const paginateGetModels = core.createPaginator(APIGatewayClient, GetModelsCommand, "position", "position", "limit");

const paginateGetResources = core.createPaginator(APIGatewayClient, GetResourcesCommand, "position", "position", "limit");

const paginateGetRestApis = core.createPaginator(APIGatewayClient, GetRestApisCommand, "position", "position", "limit");

const paginateGetUsage = core.createPaginator(APIGatewayClient, GetUsageCommand, "position", "position", "limit");

const paginateGetUsagePlanKeys = core.createPaginator(APIGatewayClient, GetUsagePlanKeysCommand, "position", "position", "limit");

const paginateGetUsagePlans = core.createPaginator(APIGatewayClient, GetUsagePlansCommand, "position", "position", "limit");

const paginateGetVpcLinks = core.createPaginator(APIGatewayClient, GetVpcLinksCommand, "position", "position", "limit");

const AccessAssociationSourceType = {
    VPCE: "VPCE",
};
const ApiKeysFormat = {
    csv: "csv",
};
const ApiKeySourceType = {
    AUTHORIZER: "AUTHORIZER",
    HEADER: "HEADER",
};
const ApiStatus = {
    AVAILABLE: "AVAILABLE",
    FAILED: "FAILED",
    PENDING: "PENDING",
    UPDATING: "UPDATING",
};
const AuthorizerType = {
    COGNITO_USER_POOLS: "COGNITO_USER_POOLS",
    REQUEST: "REQUEST",
    TOKEN: "TOKEN",
};
const CacheClusterSize = {
    SIZE_0_POINT_5_GB: "0.5",
    SIZE_118_GB: "118",
    SIZE_13_POINT_5_GB: "13.5",
    SIZE_1_POINT_6_GB: "1.6",
    SIZE_237_GB: "237",
    SIZE_28_POINT_4_GB: "28.4",
    SIZE_58_POINT_2_GB: "58.2",
    SIZE_6_POINT_1_GB: "6.1",
};
const DocumentationPartType = {
    API: "API",
    AUTHORIZER: "AUTHORIZER",
    METHOD: "METHOD",
    MODEL: "MODEL",
    PATH_PARAMETER: "PATH_PARAMETER",
    QUERY_PARAMETER: "QUERY_PARAMETER",
    REQUEST_BODY: "REQUEST_BODY",
    REQUEST_HEADER: "REQUEST_HEADER",
    RESOURCE: "RESOURCE",
    RESPONSE: "RESPONSE",
    RESPONSE_BODY: "RESPONSE_BODY",
    RESPONSE_HEADER: "RESPONSE_HEADER",
};
const EndpointAccessMode = {
    BASIC: "BASIC",
    STRICT: "STRICT",
};
const IpAddressType = {
    dualstack: "dualstack",
    ipv4: "ipv4",
};
const EndpointType = {
    EDGE: "EDGE",
    PRIVATE: "PRIVATE",
    REGIONAL: "REGIONAL",
};
const RoutingMode = {
    BASE_PATH_MAPPING_ONLY: "BASE_PATH_MAPPING_ONLY",
    ROUTING_RULE_ONLY: "ROUTING_RULE_ONLY",
    ROUTING_RULE_THEN_BASE_PATH_MAPPING: "ROUTING_RULE_THEN_BASE_PATH_MAPPING",
};
const SecurityPolicy = {
    SecurityPolicy_TLS12_2018_EDGE: "SecurityPolicy_TLS12_2018_EDGE",
    SecurityPolicy_TLS12_PFS_2025_EDGE: "SecurityPolicy_TLS12_PFS_2025_EDGE",
    SecurityPolicy_TLS13_1_2_2021_06: "SecurityPolicy_TLS13_1_2_2021_06",
    SecurityPolicy_TLS13_1_2_FIPS_PQ_2025_09: "SecurityPolicy_TLS13_1_2_FIPS_PQ_2025_09",
    SecurityPolicy_TLS13_1_2_PFS_PQ_2025_09: "SecurityPolicy_TLS13_1_2_PFS_PQ_2025_09",
    SecurityPolicy_TLS13_1_2_PQ_2025_09: "SecurityPolicy_TLS13_1_2_PQ_2025_09",
    SecurityPolicy_TLS13_1_3_2025_09: "SecurityPolicy_TLS13_1_3_2025_09",
    SecurityPolicy_TLS13_1_3_FIPS_2025_09: "SecurityPolicy_TLS13_1_3_FIPS_2025_09",
    SecurityPolicy_TLS13_2025_EDGE: "SecurityPolicy_TLS13_2025_EDGE",
    TLS_1_0: "TLS_1_0",
    TLS_1_2: "TLS_1_2",
};
const DomainNameStatus = {
    AVAILABLE: "AVAILABLE",
    FAILED: "FAILED",
    PENDING: "PENDING",
    PENDING_CERTIFICATE_REIMPORT: "PENDING_CERTIFICATE_REIMPORT",
    PENDING_OWNERSHIP_VERIFICATION: "PENDING_OWNERSHIP_VERIFICATION",
    UPDATING: "UPDATING",
};
const ConnectionType = {
    INTERNET: "INTERNET",
    VPC_LINK: "VPC_LINK",
};
const ContentHandlingStrategy = {
    CONVERT_TO_BINARY: "CONVERT_TO_BINARY",
    CONVERT_TO_TEXT: "CONVERT_TO_TEXT",
};
const ResponseTransferMode = {
    BUFFERED: "BUFFERED",
    STREAM: "STREAM",
};
const IntegrationType = {
    AWS: "AWS",
    AWS_PROXY: "AWS_PROXY",
    HTTP: "HTTP",
    HTTP_PROXY: "HTTP_PROXY",
    MOCK: "MOCK",
};
const CacheClusterStatus = {
    AVAILABLE: "AVAILABLE",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
    FLUSH_IN_PROGRESS: "FLUSH_IN_PROGRESS",
    NOT_AVAILABLE: "NOT_AVAILABLE",
};
const UnauthorizedCacheControlHeaderStrategy = {
    FAIL_WITH_403: "FAIL_WITH_403",
    SUCCEED_WITHOUT_RESPONSE_HEADER: "SUCCEED_WITHOUT_RESPONSE_HEADER",
    SUCCEED_WITH_RESPONSE_HEADER: "SUCCEED_WITH_RESPONSE_HEADER",
};
const QuotaPeriodType = {
    DAY: "DAY",
    MONTH: "MONTH",
    WEEK: "WEEK",
};
const VpcLinkStatus = {
    AVAILABLE: "AVAILABLE",
    DELETING: "DELETING",
    FAILED: "FAILED",
    PENDING: "PENDING",
};
const GatewayResponseType = {
    ACCESS_DENIED: "ACCESS_DENIED",
    API_CONFIGURATION_ERROR: "API_CONFIGURATION_ERROR",
    AUTHORIZER_CONFIGURATION_ERROR: "AUTHORIZER_CONFIGURATION_ERROR",
    AUTHORIZER_FAILURE: "AUTHORIZER_FAILURE",
    BAD_REQUEST_BODY: "BAD_REQUEST_BODY",
    BAD_REQUEST_PARAMETERS: "BAD_REQUEST_PARAMETERS",
    DEFAULT_4XX: "DEFAULT_4XX",
    DEFAULT_5XX: "DEFAULT_5XX",
    EXPIRED_TOKEN: "EXPIRED_TOKEN",
    INTEGRATION_FAILURE: "INTEGRATION_FAILURE",
    INTEGRATION_TIMEOUT: "INTEGRATION_TIMEOUT",
    INVALID_API_KEY: "INVALID_API_KEY",
    INVALID_SIGNATURE: "INVALID_SIGNATURE",
    MISSING_AUTHENTICATION_TOKEN: "MISSING_AUTHENTICATION_TOKEN",
    QUOTA_EXCEEDED: "QUOTA_EXCEEDED",
    REQUEST_TOO_LARGE: "REQUEST_TOO_LARGE",
    RESOURCE_NOT_FOUND: "RESOURCE_NOT_FOUND",
    THROTTLED: "THROTTLED",
    UNAUTHORIZED: "UNAUTHORIZED",
    UNSUPPORTED_MEDIA_TYPE: "UNSUPPORTED_MEDIA_TYPE",
    WAF_FILTERED: "WAF_FILTERED",
};
const LocationStatusType = {
    DOCUMENTED: "DOCUMENTED",
    UNDOCUMENTED: "UNDOCUMENTED",
};
const ResourceOwner = {
    OTHER_ACCOUNTS: "OTHER_ACCOUNTS",
    SELF: "SELF",
};
const PutMode = {
    Merge: "merge",
    Overwrite: "overwrite",
};
const Op = {
    add: "add",
    copy: "copy",
    move: "move",
    remove: "remove",
    replace: "replace",
    test: "test",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.APIGateway = APIGateway;
exports.APIGatewayClient = APIGatewayClient;
exports.APIGatewayServiceException = APIGatewayServiceException;
exports.APIGatewayServiceException$ = APIGatewayServiceException$;
exports.AccessAssociationSourceType = AccessAssociationSourceType;
exports.AccessLogSettings$ = AccessLogSettings$;
exports.Account$ = Account$;
exports.ApiKey$ = ApiKey$;
exports.ApiKeyIds$ = ApiKeyIds$;
exports.ApiKeySourceType = ApiKeySourceType;
exports.ApiKeys$ = ApiKeys$;
exports.ApiKeysFormat = ApiKeysFormat;
exports.ApiStage$ = ApiStage$;
exports.ApiStatus = ApiStatus;
exports.Authorizer$ = Authorizer$;
exports.AuthorizerType = AuthorizerType;
exports.Authorizers$ = Authorizers$;
exports.BadRequestException = BadRequestException;
exports.BadRequestException$ = BadRequestException$;
exports.BasePathMapping$ = BasePathMapping$;
exports.BasePathMappings$ = BasePathMappings$;
exports.CacheClusterSize = CacheClusterSize;
exports.CacheClusterStatus = CacheClusterStatus;
exports.CanarySettings$ = CanarySettings$;
exports.ClientCertificate$ = ClientCertificate$;
exports.ClientCertificates$ = ClientCertificates$;
exports.ConflictException = ConflictException;
exports.ConflictException$ = ConflictException$;
exports.ConnectionType = ConnectionType;
exports.ContentHandlingStrategy = ContentHandlingStrategy;
exports.CreateApiKey$ = CreateApiKey$;
exports.CreateApiKeyCommand = CreateApiKeyCommand;
exports.CreateApiKeyRequest$ = CreateApiKeyRequest$;
exports.CreateAuthorizer$ = CreateAuthorizer$;
exports.CreateAuthorizerCommand = CreateAuthorizerCommand;
exports.CreateAuthorizerRequest$ = CreateAuthorizerRequest$;
exports.CreateBasePathMapping$ = CreateBasePathMapping$;
exports.CreateBasePathMappingCommand = CreateBasePathMappingCommand;
exports.CreateBasePathMappingRequest$ = CreateBasePathMappingRequest$;
exports.CreateDeployment$ = CreateDeployment$;
exports.CreateDeploymentCommand = CreateDeploymentCommand;
exports.CreateDeploymentRequest$ = CreateDeploymentRequest$;
exports.CreateDocumentationPart$ = CreateDocumentationPart$;
exports.CreateDocumentationPartCommand = CreateDocumentationPartCommand;
exports.CreateDocumentationPartRequest$ = CreateDocumentationPartRequest$;
exports.CreateDocumentationVersion$ = CreateDocumentationVersion$;
exports.CreateDocumentationVersionCommand = CreateDocumentationVersionCommand;
exports.CreateDocumentationVersionRequest$ = CreateDocumentationVersionRequest$;
exports.CreateDomainName$ = CreateDomainName$;
exports.CreateDomainNameAccessAssociation$ = CreateDomainNameAccessAssociation$;
exports.CreateDomainNameAccessAssociationCommand = CreateDomainNameAccessAssociationCommand;
exports.CreateDomainNameAccessAssociationRequest$ = CreateDomainNameAccessAssociationRequest$;
exports.CreateDomainNameCommand = CreateDomainNameCommand;
exports.CreateDomainNameRequest$ = CreateDomainNameRequest$;
exports.CreateModel$ = CreateModel$;
exports.CreateModelCommand = CreateModelCommand;
exports.CreateModelRequest$ = CreateModelRequest$;
exports.CreateRequestValidator$ = CreateRequestValidator$;
exports.CreateRequestValidatorCommand = CreateRequestValidatorCommand;
exports.CreateRequestValidatorRequest$ = CreateRequestValidatorRequest$;
exports.CreateResource$ = CreateResource$;
exports.CreateResourceCommand = CreateResourceCommand;
exports.CreateResourceRequest$ = CreateResourceRequest$;
exports.CreateRestApi$ = CreateRestApi$;
exports.CreateRestApiCommand = CreateRestApiCommand;
exports.CreateRestApiRequest$ = CreateRestApiRequest$;
exports.CreateStage$ = CreateStage$;
exports.CreateStageCommand = CreateStageCommand;
exports.CreateStageRequest$ = CreateStageRequest$;
exports.CreateUsagePlan$ = CreateUsagePlan$;
exports.CreateUsagePlanCommand = CreateUsagePlanCommand;
exports.CreateUsagePlanKey$ = CreateUsagePlanKey$;
exports.CreateUsagePlanKeyCommand = CreateUsagePlanKeyCommand;
exports.CreateUsagePlanKeyRequest$ = CreateUsagePlanKeyRequest$;
exports.CreateUsagePlanRequest$ = CreateUsagePlanRequest$;
exports.CreateVpcLink$ = CreateVpcLink$;
exports.CreateVpcLinkCommand = CreateVpcLinkCommand;
exports.CreateVpcLinkRequest$ = CreateVpcLinkRequest$;
exports.DeleteApiKey$ = DeleteApiKey$;
exports.DeleteApiKeyCommand = DeleteApiKeyCommand;
exports.DeleteApiKeyRequest$ = DeleteApiKeyRequest$;
exports.DeleteAuthorizer$ = DeleteAuthorizer$;
exports.DeleteAuthorizerCommand = DeleteAuthorizerCommand;
exports.DeleteAuthorizerRequest$ = DeleteAuthorizerRequest$;
exports.DeleteBasePathMapping$ = DeleteBasePathMapping$;
exports.DeleteBasePathMappingCommand = DeleteBasePathMappingCommand;
exports.DeleteBasePathMappingRequest$ = DeleteBasePathMappingRequest$;
exports.DeleteClientCertificate$ = DeleteClientCertificate$;
exports.DeleteClientCertificateCommand = DeleteClientCertificateCommand;
exports.DeleteClientCertificateRequest$ = DeleteClientCertificateRequest$;
exports.DeleteDeployment$ = DeleteDeployment$;
exports.DeleteDeploymentCommand = DeleteDeploymentCommand;
exports.DeleteDeploymentRequest$ = DeleteDeploymentRequest$;
exports.DeleteDocumentationPart$ = DeleteDocumentationPart$;
exports.DeleteDocumentationPartCommand = DeleteDocumentationPartCommand;
exports.DeleteDocumentationPartRequest$ = DeleteDocumentationPartRequest$;
exports.DeleteDocumentationVersion$ = DeleteDocumentationVersion$;
exports.DeleteDocumentationVersionCommand = DeleteDocumentationVersionCommand;
exports.DeleteDocumentationVersionRequest$ = DeleteDocumentationVersionRequest$;
exports.DeleteDomainName$ = DeleteDomainName$;
exports.DeleteDomainNameAccessAssociation$ = DeleteDomainNameAccessAssociation$;
exports.DeleteDomainNameAccessAssociationCommand = DeleteDomainNameAccessAssociationCommand;
exports.DeleteDomainNameAccessAssociationRequest$ = DeleteDomainNameAccessAssociationRequest$;
exports.DeleteDomainNameCommand = DeleteDomainNameCommand;
exports.DeleteDomainNameRequest$ = DeleteDomainNameRequest$;
exports.DeleteGatewayResponse$ = DeleteGatewayResponse$;
exports.DeleteGatewayResponseCommand = DeleteGatewayResponseCommand;
exports.DeleteGatewayResponseRequest$ = DeleteGatewayResponseRequest$;
exports.DeleteIntegration$ = DeleteIntegration$;
exports.DeleteIntegrationCommand = DeleteIntegrationCommand;
exports.DeleteIntegrationRequest$ = DeleteIntegrationRequest$;
exports.DeleteIntegrationResponse$ = DeleteIntegrationResponse$;
exports.DeleteIntegrationResponseCommand = DeleteIntegrationResponseCommand;
exports.DeleteIntegrationResponseRequest$ = DeleteIntegrationResponseRequest$;
exports.DeleteMethod$ = DeleteMethod$;
exports.DeleteMethodCommand = DeleteMethodCommand;
exports.DeleteMethodRequest$ = DeleteMethodRequest$;
exports.DeleteMethodResponse$ = DeleteMethodResponse$;
exports.DeleteMethodResponseCommand = DeleteMethodResponseCommand;
exports.DeleteMethodResponseRequest$ = DeleteMethodResponseRequest$;
exports.DeleteModel$ = DeleteModel$;
exports.DeleteModelCommand = DeleteModelCommand;
exports.DeleteModelRequest$ = DeleteModelRequest$;
exports.DeleteRequestValidator$ = DeleteRequestValidator$;
exports.DeleteRequestValidatorCommand = DeleteRequestValidatorCommand;
exports.DeleteRequestValidatorRequest$ = DeleteRequestValidatorRequest$;
exports.DeleteResource$ = DeleteResource$;
exports.DeleteResourceCommand = DeleteResourceCommand;
exports.DeleteResourceRequest$ = DeleteResourceRequest$;
exports.DeleteRestApi$ = DeleteRestApi$;
exports.DeleteRestApiCommand = DeleteRestApiCommand;
exports.DeleteRestApiRequest$ = DeleteRestApiRequest$;
exports.DeleteStage$ = DeleteStage$;
exports.DeleteStageCommand = DeleteStageCommand;
exports.DeleteStageRequest$ = DeleteStageRequest$;
exports.DeleteUsagePlan$ = DeleteUsagePlan$;
exports.DeleteUsagePlanCommand = DeleteUsagePlanCommand;
exports.DeleteUsagePlanKey$ = DeleteUsagePlanKey$;
exports.DeleteUsagePlanKeyCommand = DeleteUsagePlanKeyCommand;
exports.DeleteUsagePlanKeyRequest$ = DeleteUsagePlanKeyRequest$;
exports.DeleteUsagePlanRequest$ = DeleteUsagePlanRequest$;
exports.DeleteVpcLink$ = DeleteVpcLink$;
exports.DeleteVpcLinkCommand = DeleteVpcLinkCommand;
exports.DeleteVpcLinkRequest$ = DeleteVpcLinkRequest$;
exports.Deployment$ = Deployment$;
exports.DeploymentCanarySettings$ = DeploymentCanarySettings$;
exports.Deployments$ = Deployments$;
exports.DocumentationPart$ = DocumentationPart$;
exports.DocumentationPartIds$ = DocumentationPartIds$;
exports.DocumentationPartLocation$ = DocumentationPartLocation$;
exports.DocumentationPartType = DocumentationPartType;
exports.DocumentationParts$ = DocumentationParts$;
exports.DocumentationVersion$ = DocumentationVersion$;
exports.DocumentationVersions$ = DocumentationVersions$;
exports.DomainName$ = DomainName$;
exports.DomainNameAccessAssociation$ = DomainNameAccessAssociation$;
exports.DomainNameAccessAssociations$ = DomainNameAccessAssociations$;
exports.DomainNameStatus = DomainNameStatus;
exports.DomainNames$ = DomainNames$;
exports.EndpointAccessMode = EndpointAccessMode;
exports.EndpointConfiguration$ = EndpointConfiguration$;
exports.EndpointType = EndpointType;
exports.ExportResponse$ = ExportResponse$;
exports.FlushStageAuthorizersCache$ = FlushStageAuthorizersCache$;
exports.FlushStageAuthorizersCacheCommand = FlushStageAuthorizersCacheCommand;
exports.FlushStageAuthorizersCacheRequest$ = FlushStageAuthorizersCacheRequest$;
exports.FlushStageCache$ = FlushStageCache$;
exports.FlushStageCacheCommand = FlushStageCacheCommand;
exports.FlushStageCacheRequest$ = FlushStageCacheRequest$;
exports.GatewayResponse$ = GatewayResponse$;
exports.GatewayResponseType = GatewayResponseType;
exports.GatewayResponses$ = GatewayResponses$;
exports.GenerateClientCertificate$ = GenerateClientCertificate$;
exports.GenerateClientCertificateCommand = GenerateClientCertificateCommand;
exports.GenerateClientCertificateRequest$ = GenerateClientCertificateRequest$;
exports.GetAccount$ = GetAccount$;
exports.GetAccountCommand = GetAccountCommand;
exports.GetAccountRequest$ = GetAccountRequest$;
exports.GetApiKey$ = GetApiKey$;
exports.GetApiKeyCommand = GetApiKeyCommand;
exports.GetApiKeyRequest$ = GetApiKeyRequest$;
exports.GetApiKeys$ = GetApiKeys$;
exports.GetApiKeysCommand = GetApiKeysCommand;
exports.GetApiKeysRequest$ = GetApiKeysRequest$;
exports.GetAuthorizer$ = GetAuthorizer$;
exports.GetAuthorizerCommand = GetAuthorizerCommand;
exports.GetAuthorizerRequest$ = GetAuthorizerRequest$;
exports.GetAuthorizers$ = GetAuthorizers$;
exports.GetAuthorizersCommand = GetAuthorizersCommand;
exports.GetAuthorizersRequest$ = GetAuthorizersRequest$;
exports.GetBasePathMapping$ = GetBasePathMapping$;
exports.GetBasePathMappingCommand = GetBasePathMappingCommand;
exports.GetBasePathMappingRequest$ = GetBasePathMappingRequest$;
exports.GetBasePathMappings$ = GetBasePathMappings$;
exports.GetBasePathMappingsCommand = GetBasePathMappingsCommand;
exports.GetBasePathMappingsRequest$ = GetBasePathMappingsRequest$;
exports.GetClientCertificate$ = GetClientCertificate$;
exports.GetClientCertificateCommand = GetClientCertificateCommand;
exports.GetClientCertificateRequest$ = GetClientCertificateRequest$;
exports.GetClientCertificates$ = GetClientCertificates$;
exports.GetClientCertificatesCommand = GetClientCertificatesCommand;
exports.GetClientCertificatesRequest$ = GetClientCertificatesRequest$;
exports.GetDeployment$ = GetDeployment$;
exports.GetDeploymentCommand = GetDeploymentCommand;
exports.GetDeploymentRequest$ = GetDeploymentRequest$;
exports.GetDeployments$ = GetDeployments$;
exports.GetDeploymentsCommand = GetDeploymentsCommand;
exports.GetDeploymentsRequest$ = GetDeploymentsRequest$;
exports.GetDocumentationPart$ = GetDocumentationPart$;
exports.GetDocumentationPartCommand = GetDocumentationPartCommand;
exports.GetDocumentationPartRequest$ = GetDocumentationPartRequest$;
exports.GetDocumentationParts$ = GetDocumentationParts$;
exports.GetDocumentationPartsCommand = GetDocumentationPartsCommand;
exports.GetDocumentationPartsRequest$ = GetDocumentationPartsRequest$;
exports.GetDocumentationVersion$ = GetDocumentationVersion$;
exports.GetDocumentationVersionCommand = GetDocumentationVersionCommand;
exports.GetDocumentationVersionRequest$ = GetDocumentationVersionRequest$;
exports.GetDocumentationVersions$ = GetDocumentationVersions$;
exports.GetDocumentationVersionsCommand = GetDocumentationVersionsCommand;
exports.GetDocumentationVersionsRequest$ = GetDocumentationVersionsRequest$;
exports.GetDomainName$ = GetDomainName$;
exports.GetDomainNameAccessAssociations$ = GetDomainNameAccessAssociations$;
exports.GetDomainNameAccessAssociationsCommand = GetDomainNameAccessAssociationsCommand;
exports.GetDomainNameAccessAssociationsRequest$ = GetDomainNameAccessAssociationsRequest$;
exports.GetDomainNameCommand = GetDomainNameCommand;
exports.GetDomainNameRequest$ = GetDomainNameRequest$;
exports.GetDomainNames$ = GetDomainNames$;
exports.GetDomainNamesCommand = GetDomainNamesCommand;
exports.GetDomainNamesRequest$ = GetDomainNamesRequest$;
exports.GetExport$ = GetExport$;
exports.GetExportCommand = GetExportCommand;
exports.GetExportRequest$ = GetExportRequest$;
exports.GetGatewayResponse$ = GetGatewayResponse$;
exports.GetGatewayResponseCommand = GetGatewayResponseCommand;
exports.GetGatewayResponseRequest$ = GetGatewayResponseRequest$;
exports.GetGatewayResponses$ = GetGatewayResponses$;
exports.GetGatewayResponsesCommand = GetGatewayResponsesCommand;
exports.GetGatewayResponsesRequest$ = GetGatewayResponsesRequest$;
exports.GetIntegration$ = GetIntegration$;
exports.GetIntegrationCommand = GetIntegrationCommand;
exports.GetIntegrationRequest$ = GetIntegrationRequest$;
exports.GetIntegrationResponse$ = GetIntegrationResponse$;
exports.GetIntegrationResponseCommand = GetIntegrationResponseCommand;
exports.GetIntegrationResponseRequest$ = GetIntegrationResponseRequest$;
exports.GetMethod$ = GetMethod$;
exports.GetMethodCommand = GetMethodCommand;
exports.GetMethodRequest$ = GetMethodRequest$;
exports.GetMethodResponse$ = GetMethodResponse$;
exports.GetMethodResponseCommand = GetMethodResponseCommand;
exports.GetMethodResponseRequest$ = GetMethodResponseRequest$;
exports.GetModel$ = GetModel$;
exports.GetModelCommand = GetModelCommand;
exports.GetModelRequest$ = GetModelRequest$;
exports.GetModelTemplate$ = GetModelTemplate$;
exports.GetModelTemplateCommand = GetModelTemplateCommand;
exports.GetModelTemplateRequest$ = GetModelTemplateRequest$;
exports.GetModels$ = GetModels$;
exports.GetModelsCommand = GetModelsCommand;
exports.GetModelsRequest$ = GetModelsRequest$;
exports.GetRequestValidator$ = GetRequestValidator$;
exports.GetRequestValidatorCommand = GetRequestValidatorCommand;
exports.GetRequestValidatorRequest$ = GetRequestValidatorRequest$;
exports.GetRequestValidators$ = GetRequestValidators$;
exports.GetRequestValidatorsCommand = GetRequestValidatorsCommand;
exports.GetRequestValidatorsRequest$ = GetRequestValidatorsRequest$;
exports.GetResource$ = GetResource$;
exports.GetResourceCommand = GetResourceCommand;
exports.GetResourceRequest$ = GetResourceRequest$;
exports.GetResources$ = GetResources$;
exports.GetResourcesCommand = GetResourcesCommand;
exports.GetResourcesRequest$ = GetResourcesRequest$;
exports.GetRestApi$ = GetRestApi$;
exports.GetRestApiCommand = GetRestApiCommand;
exports.GetRestApiRequest$ = GetRestApiRequest$;
exports.GetRestApis$ = GetRestApis$;
exports.GetRestApisCommand = GetRestApisCommand;
exports.GetRestApisRequest$ = GetRestApisRequest$;
exports.GetSdk$ = GetSdk$;
exports.GetSdkCommand = GetSdkCommand;
exports.GetSdkRequest$ = GetSdkRequest$;
exports.GetSdkType$ = GetSdkType$;
exports.GetSdkTypeCommand = GetSdkTypeCommand;
exports.GetSdkTypeRequest$ = GetSdkTypeRequest$;
exports.GetSdkTypes$ = GetSdkTypes$;
exports.GetSdkTypesCommand = GetSdkTypesCommand;
exports.GetSdkTypesRequest$ = GetSdkTypesRequest$;
exports.GetStage$ = GetStage$;
exports.GetStageCommand = GetStageCommand;
exports.GetStageRequest$ = GetStageRequest$;
exports.GetStages$ = GetStages$;
exports.GetStagesCommand = GetStagesCommand;
exports.GetStagesRequest$ = GetStagesRequest$;
exports.GetTags$ = GetTags$;
exports.GetTagsCommand = GetTagsCommand;
exports.GetTagsRequest$ = GetTagsRequest$;
exports.GetUsage$ = GetUsage$;
exports.GetUsageCommand = GetUsageCommand;
exports.GetUsagePlan$ = GetUsagePlan$;
exports.GetUsagePlanCommand = GetUsagePlanCommand;
exports.GetUsagePlanKey$ = GetUsagePlanKey$;
exports.GetUsagePlanKeyCommand = GetUsagePlanKeyCommand;
exports.GetUsagePlanKeyRequest$ = GetUsagePlanKeyRequest$;
exports.GetUsagePlanKeys$ = GetUsagePlanKeys$;
exports.GetUsagePlanKeysCommand = GetUsagePlanKeysCommand;
exports.GetUsagePlanKeysRequest$ = GetUsagePlanKeysRequest$;
exports.GetUsagePlanRequest$ = GetUsagePlanRequest$;
exports.GetUsagePlans$ = GetUsagePlans$;
exports.GetUsagePlansCommand = GetUsagePlansCommand;
exports.GetUsagePlansRequest$ = GetUsagePlansRequest$;
exports.GetUsageRequest$ = GetUsageRequest$;
exports.GetVpcLink$ = GetVpcLink$;
exports.GetVpcLinkCommand = GetVpcLinkCommand;
exports.GetVpcLinkRequest$ = GetVpcLinkRequest$;
exports.GetVpcLinks$ = GetVpcLinks$;
exports.GetVpcLinksCommand = GetVpcLinksCommand;
exports.GetVpcLinksRequest$ = GetVpcLinksRequest$;
exports.ImportApiKeys$ = ImportApiKeys$;
exports.ImportApiKeysCommand = ImportApiKeysCommand;
exports.ImportApiKeysRequest$ = ImportApiKeysRequest$;
exports.ImportDocumentationParts$ = ImportDocumentationParts$;
exports.ImportDocumentationPartsCommand = ImportDocumentationPartsCommand;
exports.ImportDocumentationPartsRequest$ = ImportDocumentationPartsRequest$;
exports.ImportRestApi$ = ImportRestApi$;
exports.ImportRestApiCommand = ImportRestApiCommand;
exports.ImportRestApiRequest$ = ImportRestApiRequest$;
exports.Integration$ = Integration$;
exports.IntegrationResponse$ = IntegrationResponse$;
exports.IntegrationType = IntegrationType;
exports.IpAddressType = IpAddressType;
exports.LimitExceededException = LimitExceededException;
exports.LimitExceededException$ = LimitExceededException$;
exports.LocationStatusType = LocationStatusType;
exports.Method$ = Method$;
exports.MethodResponse$ = MethodResponse$;
exports.MethodSetting$ = MethodSetting$;
exports.MethodSnapshot$ = MethodSnapshot$;
exports.Model$ = Model$;
exports.Models$ = Models$;
exports.MutualTlsAuthentication$ = MutualTlsAuthentication$;
exports.MutualTlsAuthenticationInput$ = MutualTlsAuthenticationInput$;
exports.NotFoundException = NotFoundException;
exports.NotFoundException$ = NotFoundException$;
exports.Op = Op;
exports.PatchOperation$ = PatchOperation$;
exports.PutGatewayResponse$ = PutGatewayResponse$;
exports.PutGatewayResponseCommand = PutGatewayResponseCommand;
exports.PutGatewayResponseRequest$ = PutGatewayResponseRequest$;
exports.PutIntegration$ = PutIntegration$;
exports.PutIntegrationCommand = PutIntegrationCommand;
exports.PutIntegrationRequest$ = PutIntegrationRequest$;
exports.PutIntegrationResponse$ = PutIntegrationResponse$;
exports.PutIntegrationResponseCommand = PutIntegrationResponseCommand;
exports.PutIntegrationResponseRequest$ = PutIntegrationResponseRequest$;
exports.PutMethod$ = PutMethod$;
exports.PutMethodCommand = PutMethodCommand;
exports.PutMethodRequest$ = PutMethodRequest$;
exports.PutMethodResponse$ = PutMethodResponse$;
exports.PutMethodResponseCommand = PutMethodResponseCommand;
exports.PutMethodResponseRequest$ = PutMethodResponseRequest$;
exports.PutMode = PutMode;
exports.PutRestApi$ = PutRestApi$;
exports.PutRestApiCommand = PutRestApiCommand;
exports.PutRestApiRequest$ = PutRestApiRequest$;
exports.QuotaPeriodType = QuotaPeriodType;
exports.QuotaSettings$ = QuotaSettings$;
exports.RejectDomainNameAccessAssociation$ = RejectDomainNameAccessAssociation$;
exports.RejectDomainNameAccessAssociationCommand = RejectDomainNameAccessAssociationCommand;
exports.RejectDomainNameAccessAssociationRequest$ = RejectDomainNameAccessAssociationRequest$;
exports.RequestValidator$ = RequestValidator$;
exports.RequestValidators$ = RequestValidators$;
exports.Resource$ = Resource$;
exports.ResourceOwner = ResourceOwner;
exports.Resources$ = Resources$;
exports.ResponseTransferMode = ResponseTransferMode;
exports.RestApi$ = RestApi$;
exports.RestApis$ = RestApis$;
exports.RoutingMode = RoutingMode;
exports.SdkConfigurationProperty$ = SdkConfigurationProperty$;
exports.SdkResponse$ = SdkResponse$;
exports.SdkType$ = SdkType$;
exports.SdkTypes$ = SdkTypes$;
exports.SecurityPolicy = SecurityPolicy;
exports.ServiceUnavailableException = ServiceUnavailableException;
exports.ServiceUnavailableException$ = ServiceUnavailableException$;
exports.Stage$ = Stage$;
exports.StageKey$ = StageKey$;
exports.Stages$ = Stages$;
exports.TagResource$ = TagResource$;
exports.TagResourceCommand = TagResourceCommand;
exports.TagResourceRequest$ = TagResourceRequest$;
exports.Tags$ = Tags$;
exports.Template$ = Template$;
exports.TestInvokeAuthorizer$ = TestInvokeAuthorizer$;
exports.TestInvokeAuthorizerCommand = TestInvokeAuthorizerCommand;
exports.TestInvokeAuthorizerRequest$ = TestInvokeAuthorizerRequest$;
exports.TestInvokeAuthorizerResponse$ = TestInvokeAuthorizerResponse$;
exports.TestInvokeMethod$ = TestInvokeMethod$;
exports.TestInvokeMethodCommand = TestInvokeMethodCommand;
exports.TestInvokeMethodRequest$ = TestInvokeMethodRequest$;
exports.TestInvokeMethodResponse$ = TestInvokeMethodResponse$;
exports.ThrottleSettings$ = ThrottleSettings$;
exports.TlsConfig$ = TlsConfig$;
exports.TooManyRequestsException = TooManyRequestsException;
exports.TooManyRequestsException$ = TooManyRequestsException$;
exports.UnauthorizedCacheControlHeaderStrategy = UnauthorizedCacheControlHeaderStrategy;
exports.UnauthorizedException = UnauthorizedException;
exports.UnauthorizedException$ = UnauthorizedException$;
exports.UntagResource$ = UntagResource$;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UntagResourceRequest$ = UntagResourceRequest$;
exports.UpdateAccount$ = UpdateAccount$;
exports.UpdateAccountCommand = UpdateAccountCommand;
exports.UpdateAccountRequest$ = UpdateAccountRequest$;
exports.UpdateApiKey$ = UpdateApiKey$;
exports.UpdateApiKeyCommand = UpdateApiKeyCommand;
exports.UpdateApiKeyRequest$ = UpdateApiKeyRequest$;
exports.UpdateAuthorizer$ = UpdateAuthorizer$;
exports.UpdateAuthorizerCommand = UpdateAuthorizerCommand;
exports.UpdateAuthorizerRequest$ = UpdateAuthorizerRequest$;
exports.UpdateBasePathMapping$ = UpdateBasePathMapping$;
exports.UpdateBasePathMappingCommand = UpdateBasePathMappingCommand;
exports.UpdateBasePathMappingRequest$ = UpdateBasePathMappingRequest$;
exports.UpdateClientCertificate$ = UpdateClientCertificate$;
exports.UpdateClientCertificateCommand = UpdateClientCertificateCommand;
exports.UpdateClientCertificateRequest$ = UpdateClientCertificateRequest$;
exports.UpdateDeployment$ = UpdateDeployment$;
exports.UpdateDeploymentCommand = UpdateDeploymentCommand;
exports.UpdateDeploymentRequest$ = UpdateDeploymentRequest$;
exports.UpdateDocumentationPart$ = UpdateDocumentationPart$;
exports.UpdateDocumentationPartCommand = UpdateDocumentationPartCommand;
exports.UpdateDocumentationPartRequest$ = UpdateDocumentationPartRequest$;
exports.UpdateDocumentationVersion$ = UpdateDocumentationVersion$;
exports.UpdateDocumentationVersionCommand = UpdateDocumentationVersionCommand;
exports.UpdateDocumentationVersionRequest$ = UpdateDocumentationVersionRequest$;
exports.UpdateDomainName$ = UpdateDomainName$;
exports.UpdateDomainNameCommand = UpdateDomainNameCommand;
exports.UpdateDomainNameRequest$ = UpdateDomainNameRequest$;
exports.UpdateGatewayResponse$ = UpdateGatewayResponse$;
exports.UpdateGatewayResponseCommand = UpdateGatewayResponseCommand;
exports.UpdateGatewayResponseRequest$ = UpdateGatewayResponseRequest$;
exports.UpdateIntegration$ = UpdateIntegration$;
exports.UpdateIntegrationCommand = UpdateIntegrationCommand;
exports.UpdateIntegrationRequest$ = UpdateIntegrationRequest$;
exports.UpdateIntegrationResponse$ = UpdateIntegrationResponse$;
exports.UpdateIntegrationResponseCommand = UpdateIntegrationResponseCommand;
exports.UpdateIntegrationResponseRequest$ = UpdateIntegrationResponseRequest$;
exports.UpdateMethod$ = UpdateMethod$;
exports.UpdateMethodCommand = UpdateMethodCommand;
exports.UpdateMethodRequest$ = UpdateMethodRequest$;
exports.UpdateMethodResponse$ = UpdateMethodResponse$;
exports.UpdateMethodResponseCommand = UpdateMethodResponseCommand;
exports.UpdateMethodResponseRequest$ = UpdateMethodResponseRequest$;
exports.UpdateModel$ = UpdateModel$;
exports.UpdateModelCommand = UpdateModelCommand;
exports.UpdateModelRequest$ = UpdateModelRequest$;
exports.UpdateRequestValidator$ = UpdateRequestValidator$;
exports.UpdateRequestValidatorCommand = UpdateRequestValidatorCommand;
exports.UpdateRequestValidatorRequest$ = UpdateRequestValidatorRequest$;
exports.UpdateResource$ = UpdateResource$;
exports.UpdateResourceCommand = UpdateResourceCommand;
exports.UpdateResourceRequest$ = UpdateResourceRequest$;
exports.UpdateRestApi$ = UpdateRestApi$;
exports.UpdateRestApiCommand = UpdateRestApiCommand;
exports.UpdateRestApiRequest$ = UpdateRestApiRequest$;
exports.UpdateStage$ = UpdateStage$;
exports.UpdateStageCommand = UpdateStageCommand;
exports.UpdateStageRequest$ = UpdateStageRequest$;
exports.UpdateUsage$ = UpdateUsage$;
exports.UpdateUsageCommand = UpdateUsageCommand;
exports.UpdateUsagePlan$ = UpdateUsagePlan$;
exports.UpdateUsagePlanCommand = UpdateUsagePlanCommand;
exports.UpdateUsagePlanRequest$ = UpdateUsagePlanRequest$;
exports.UpdateUsageRequest$ = UpdateUsageRequest$;
exports.UpdateVpcLink$ = UpdateVpcLink$;
exports.UpdateVpcLinkCommand = UpdateVpcLinkCommand;
exports.UpdateVpcLinkRequest$ = UpdateVpcLinkRequest$;
exports.Usage$ = Usage$;
exports.UsagePlan$ = UsagePlan$;
exports.UsagePlanKey$ = UsagePlanKey$;
exports.UsagePlanKeys$ = UsagePlanKeys$;
exports.UsagePlans$ = UsagePlans$;
exports.VpcLink$ = VpcLink$;
exports.VpcLinkStatus = VpcLinkStatus;
exports.VpcLinks$ = VpcLinks$;
exports.paginateGetApiKeys = paginateGetApiKeys;
exports.paginateGetBasePathMappings = paginateGetBasePathMappings;
exports.paginateGetClientCertificates = paginateGetClientCertificates;
exports.paginateGetDeployments = paginateGetDeployments;
exports.paginateGetDomainNames = paginateGetDomainNames;
exports.paginateGetModels = paginateGetModels;
exports.paginateGetResources = paginateGetResources;
exports.paginateGetRestApis = paginateGetRestApis;
exports.paginateGetUsage = paginateGetUsage;
exports.paginateGetUsagePlanKeys = paginateGetUsagePlanKeys;
exports.paginateGetUsagePlans = paginateGetUsagePlans;
exports.paginateGetVpcLinks = paginateGetVpcLinks;
